<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Spatial Trajectory Analysis â€¢ SPATA2</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Spatial Trajectory Analysis">
<meta property="og:description" content="SPATA2">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SPATA2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/spata-v2-introduction-and-philosophy.html">About</a>
</li>
<li>
  <a href="../articles/spata-v2-installation.html">Installation</a>
</li>
<li>
  <a href="../articles/spata-v2-terminology.html">Terminology</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Around the Spata-Object</li>
    <li>
      <a href="../articles/spata-v2-object-initiation-and-manipulation.html">Object Set Up and Manipulation</a>
    </li>
    <li>
      <a href="../articles/spata-v2-extract-add-and-join-data.html">Extracting, Adding and Joining Data</a>
    </li>
    <li>
      <a href="../articles/spata-v2-platform-compatibility.html">Integrating Different Platforms</a>
    </li>
    <li class="dropdown-header">Analysis &amp; Computation</li>
    <li>
      <a href="../articles/spata-v2-autoencoder.html">Autoencoder Denoising</a>
    </li>
    <li>
      <a href="../articles/spata-v2-de-analysis.html">Differential Expression Analysis</a>
    </li>
    <li>
      <a href="../articles/spata-v2-segmentation-and-clustering.html">Spatial Segmentation &amp; Clustering</a>
    </li>
    <li>
      <a href="../articles/spata-v2-trajectory-analysis.html">Spatial Trajectories</a>
    </li>
    <li class="dropdown-header">Visualization</li>
    <li>
      <a href="../articles/spata-v2-plotting-surface.html">Surface Plotting</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-dimensional-reduction.html">Dimensional Reduction</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-descriptive-statistics.html">Descriptive Statistics</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-miscellaneous.html">Miscellaneous</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="spata-v2-trajectory-analysis_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Spatial Trajectory Analysis</h1>
            
      
      
      <div class="hidden name"><code>spata-v2-trajectory-analysis.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>1. Introduction</h2>
<p>With spatial trajectory analysis <strong>SPATA2</strong> introduces a new approach to find, analyze and visualize differently expressed genes and gene-sets in a spatial context. While the classic differential gene expression analyzes differences between experimental groups as a whole it neglects changes of expression levels that can only be seen while maintaining the spatial dimensions. Spatial trajectories allow to answer questions that include such a spatial component. E.g.:</p>
<ol style="list-style-type: decimal">
<li>In how far do expression levels change the more we move towards a region of interest?</li>
<li>Which genes follow the same pattern along these paths?</li>
</ol>
<p>The spatial trajectory tools provided in <strong>SPATA2</strong> enable new ways of visualization as well as new possibilities to screen for genes, gene-sets and patterns of interest.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://themilolab.com/">SPATA2</a></span><span class="op">)</span>

<span class="co"># load spata-object </span>
<span class="va">spata_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/loadSpataObject.html">loadSpataObject</a></span><span class="op">(</span><span class="st">"data/spata-obj-gbm275.RDS"</span><span class="op">)</span> </code></pre></div>
</div>
<div id="draw-trajectories" class="section level2">
<h2 class="hasAnchor">
<a href="#draw-trajectories" class="anchor"></a>2. Draw trajectories</h2>
<p>Spatial trajectories of a sample in a given spata-object can be drawn interactively using the function <code><a href="../reference/createTrajectories.html">createTrajectories()</a></code>as shown in the example below. <code><a href="../reference/createTrajectories.html">createTrajectories()</a></code> opens a mini-shiny application. This app allows one the one hand to investigate the sample with regards to spatial gene expression like <code><a href="../reference/plotSurface.html">plotSurfaceInteractive()</a></code> does and on the other hand to draw trajectories through the areas of interest in four easy steps.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># open interactive application</span>
<span class="va">spata_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createTrajectories.html">createTrajectories</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<div id="drawing" class="section level3">
<h3 class="hasAnchor">
<a href="#drawing" class="anchor"></a>2.1 Drawing</h3>
<p>Draw the trajectory interactively by clicking on the locations where you want the start and endpoints to be. If you click more than two times the trajectory is divided into sub-trajectory parts which will be displayed by subsequent plotting functions. Hence, it makes sense to stop everytime where you would expect a change as this will be highlighted by the transition from part to the subsequent one.</p>
<div class="figure" style="text-align: left">
<img src="images/tumor-traj-1.PNG" alt="Figure 2.1 Interactive trajectory drawing" width="100%"><p class="caption">
Figure 2.1 Interactive trajectory drawing
</p>
</div>
</div>
<div id="highlighting" class="section level3">
<h3 class="hasAnchor">
<a href="#highlighting" class="anchor"></a>2.2 Highlighting</h3>
<p>If start- and end-points are set adjust the trajectory width to and click on â€˜Highlightâ€™.</p>
<div class="figure" style="text-align: left">
<img src="images/tumor-traj-2.PNG" alt="Figure 2.2 Highlight the trajectory" width="100%"><p class="caption">
Figure 2.2 Highlight the trajectory
</p>
</div>
</div>
<div id="naming" class="section level3">
<h3 class="hasAnchor">
<a href="#naming" class="anchor"></a>2.3 Naming</h3>
<p>If you are satisfied with your trajectory enter a valid trajectory name and - if you want - a comment.</p>
</div>
<div id="saving" class="section level3">
<h3 class="hasAnchor">
<a href="#saving" class="anchor"></a>2.4 Saving</h3>
<p>Click on â€˜Save Trajectoryâ€™ and then on â€˜Close applicationâ€™ in order to return the updated spata-object.</p>
<p>Trajectories are stored inside the object itself. Therefore <code><a href="../reference/createTrajectories.html">createTrajectories()</a></code> returns an updated version of the object that was specified as the functionâ€™s input now containing the additional trajectories.</p>
<p>To obtain all trajectory names that have been drawn and saved in your spata-object use <code><a href="../reference/getTrajectoryNames.html">getTrajectoryNames()</a></code>. Depending on the number of samples specified it returns a character vector or a list of such.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get trajectory names </span>
<span class="fu"><a href="../reference/getTrajectoryNames.html">getTrajectoryNames</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "tumor-areas"</code></pre>
</div>
</div>
<div id="visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#visualization" class="anchor"></a>3. Visualization</h2>
<div id="spatial-course" class="section level3">
<h3 class="hasAnchor">
<a href="#spatial-course" class="anchor"></a>3.1 Spatial course</h3>
<p>In order to visualize specific trajectories on the sampleâ€™s surface use of <code><a href="../reference/plotTrajectory.html">plotTrajectory()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTrajectory.html">plotTrajectory</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
               trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>,
               color_by <span class="op">=</span> <span class="st">"seurat_clusters"</span>,
               pt_clrp <span class="op">=</span> <span class="st">"npg"</span>,
               pt_alpha <span class="op">=</span> <span class="fl">0.25</span>, <span class="co"># reduce alpha to highlight the trajectory's course</span>
               display_image <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/legendBottom.html">legendTop</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="../reference/plotTrajectory.html">plotTrajectory</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
               trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>,
               color_by <span class="op">=</span> <span class="st">"nCount_Spatial"</span>,
               smooth <span class="op">=</span> <span class="cn">TRUE</span>, 
               pt_alpha <span class="op">=</span> <span class="fl">0.25</span>, 
               display_image <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/legendBottom.html">legendTop</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-trajectory-analysis_files/figure-html/visualize-trajectories-1.png" alt="Figure 3.1 Surface related visualization of the trajectory" width="50%"><img src="spata-v2-trajectory-analysis_files/figure-html/visualize-trajectories-2.png" alt="Figure 3.1 Surface related visualization of the trajectory" width="50%"><p class="caption">
Figure 3.1 Surface related visualization of the trajectory
</p>
</div>
</div>
<div id="spatial-trends-and-dynamics" class="section level3">
<h3 class="hasAnchor">
<a href="#spatial-trends-and-dynamics" class="anchor"></a>3.2 Spatial trends and dynamics</h3>
<div id="trajectory-trends-via-lineplots" class="section level4">
<h4 class="hasAnchor">
<a href="#trajectory-trends-via-lineplots" class="anchor"></a>3.2.1 Trajectory trends via lineplots</h4>
<p>To display the trend of certain variables along a trajectory via lineplots use one of the following three functions:</p>
<ul>
<li><code><a href="../reference/plotTrajectoryFeatures.html">plotTrajectoryFeatures()</a></code></li>
<li><code><a href="../reference/plotTrajectoryFeatures.html">plotTrajectoryGeneSets()</a></code></li>
<li><code><a href="../reference/plotTrajectoryFeatures.html">plotTrajectoryGenes()</a></code></li>
</ul>
<p>A call to one of those three functions will return a smoothed lineplot where the direction of the trajectory is mapped onto the x-axis and the values of the chosen aspects are mapped onto the y-axis. All functions follow the same concept and share the majority of their arguments. Still, they feature slight differences given the nature of their aspects to display.</p>
<p>The following three arguments have to be set in order to specify the trajectory that is to be analyzed:</p>
<ul>
<li><code>object</code></li>
<li><code>trajectory_name</code></li>
</ul>
<p>The way values are smoothed and displayed can be manipulated via:</p>
<ul>
<li><code>smooth_method</code></li>
<li><code>smooth_span</code></li>
<li><code>smooth_se</code></li>
</ul>
</div>
<div id="features-along-trajectories" class="section level4">
<h4 class="hasAnchor">
<a href="#features-along-trajectories" class="anchor"></a>Features along trajectories</h4>
<p>Specify the features of interest via the argument <code>features</code>. Note that the function will compute the mean of all points on the axis orthogonal to the trajectoryâ€™s direction. As only numeric variables allow for such computation displaying the trend of categorical variables (e.g.Â cluster) is not possible via lineplots.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTrajectoryFeatures.html">plotTrajectoryFeatures</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>,
                       trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>,
                       features <span class="op">=</span> <span class="st">"nCount_Spatial"</span>, 
                       smooth_method <span class="op">=</span> <span class="st">"loess"</span>, 
                       smooth_span <span class="op">=</span> <span class="fl">0.2</span>, 
                       smooth_se <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </code></pre></div>
<div class="figure">
<img src="spata-v2-trajectory-analysis_files/figure-html/trajectory-features-1-1.png" alt="Figure 3.2 Visualization of ontinuous features along a trajectory" width="100%"><p class="caption">
Figure 3.2 Visualization of ontinuous features along a trajectory
</p>
</div>
<p>In order to plot discrete features use <code><a href="../reference/plotTrajectoryFeaturesDiscrete.html">plotTrajectoryFeaturesDiscrete()</a></code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTrajectoryFeaturesDiscrete.html">plotTrajectoryFeaturesDiscrete</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>,
                               trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>,
                               discrete_feature <span class="op">=</span> <span class="st">"seurat_clusters"</span>, 
                               clrp <span class="op">=</span> <span class="st">"npg"</span>,
                               display_trajectory_parts <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </code></pre></div>
<div class="figure">
<img src="spata-v2-trajectory-analysis_files/figure-html/trajectory-features-2-1.png" alt="Figure 3.3 Visualization of discrete features along a trajectory" width="100%"><p class="caption">
Figure 3.3 Visualization of discrete features along a trajectory
</p>
</div>
</div>
<div id="genes-and-gene-sets-along-trajectories" class="section level4">
<h4 class="hasAnchor">
<a href="#genes-and-gene-sets-along-trajectories" class="anchor"></a>Genes and gene-sets along trajectories</h4>
<p>Specify the genes or gene-sets of interest by using the respective functions and arguments.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># gene-set names</span>
<span class="va">genes_of_interest</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CALM1"</span>, <span class="st">"VIM"</span>, <span class="st">"GFAP"</span>, <span class="st">"METRN"</span><span class="op">)</span>
  
<span class="co"># plot lineplot</span>
<span class="fu"><a href="../reference/plotTrajectoryFeatures.html">plotTrajectoryGenes</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>,
                    trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>, 
                    genes <span class="op">=</span> <span class="va">genes_of_interest</span>,
                    smooth_span <span class="op">=</span> <span class="fl">0.2</span>,
                    smooth_se <span class="op">=</span> <span class="cn">TRUE</span>, 
                    display_facets <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># use facet_wrap() to split the plot in four parts</span>
                    nrow <span class="op">=</span> <span class="fl">2</span> <span class="co"># align the sub plots in two rows </span>
                    <span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-trajectory-analysis_files/figure-html/trajectory-genes-1.png" alt="Figure 3.4 Visualization of gene expression along a trajectory" width="100%"><p class="caption">
Figure 3.4 Visualization of gene expression along a trajectory
</p>
</div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTrajectoryFeatures.html">plotTrajectoryGeneSets</a></span><span class="op">(</span>
  object <span class="op">=</span> <span class="va">spata_obj</span>,
  trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>,
  gene_sets <span class="op">=</span> <span class="st">"HM_HYPOXIA"</span>,
  display_trajectory_parts <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span> <span class="co"># results in missing vertical lines </span>
  <span class="fu"><a href="../reference/legendBottom.html">legendTop</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="spata-v2-trajectory-analysis_files/figure-html/trajectory-genesets-1.png" width="100%"></p>
</div>
</div>
</div>
<div id="analyze-model-trajectory-trends" class="section level2">
<h2 class="hasAnchor">
<a href="#analyze-model-trajectory-trends" class="anchor"></a>4. Analyze &amp; model trajectory trends</h2>
<p>Once a trajectory is drawn in the light of a biologic question or finding dynamics of additional variables can be easily visualized and investigated one by one. However, given that the average expression matrix contains tens of thousands of genes investigating all genes and gene-sets manually can quickly become a cumbersome task. By abstracting relevant biological dynamics into mathematical models the trajectory can be screened for variables (gene- and gene-set expression in particular) that correspond to these models.</p>
<div id="predefined-models" class="section level3">
<h3 class="hasAnchor">
<a href="#predefined-models" class="anchor"></a>4.1 Predefined models</h3>
<p>To find variables that follow a certain trend along the trajectory of interest use <code><a href="../reference/assessTrajectoryTrends.html">assessTrajectoryTrends()</a></code>. It fits the trajectory-trends of all specified variables to a variety of patterns/mathematical models (Figure 4.1) and assesses their fit by calculating the area under the curve of the residuals.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getGenes.html">getGenes</a></span><span class="op">(</span><span class="va">spata_obj</span><span class="op">)</span>
<span class="va">all_gene_sets</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getGeneSets.html">getGeneSets</a></span><span class="op">(</span><span class="va">spata_obj</span><span class="op">)</span>

<span class="co"># obtain an assessed trajectory data.frame for all genes</span>
<span class="va">atdf_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assessTrajectoryTrends.html">assessTrajectoryTrends</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
                                     trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>, 
                                     variables <span class="op">=</span> <span class="va">all_genes</span><span class="op">)</span>

<span class="co"># obtain an assessed trajectory data.frame for all gene-sets</span>
<span class="va">atdf_gene_sets</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assessTrajectoryTrends.html">assessTrajectoryTrends</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
                                         trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>, 
                                         variables <span class="op">=</span> <span class="va">all_gene</span><span class="op">-</span><span class="va">sets</span><span class="op">)</span>

<span class="co"># output example</span>
<span class="va">atdf_genes</span></code></pre></div>
<pre><code>## # A tibble: 237,538 x 3
## # Groups:   variables [16,967]
##    variables pattern               auc
##    &lt;chr&gt;     &lt;chr&gt;               &lt;dbl&gt;
##  1 MT3       Gradient ascending  0.184
##  2 GFAP      Gradient descending 0.387
##  3 MT3       Linear ascending    0.387
##  4 PDCD6     Linear descending   0.404
##  5 GFAP      Linear descending   0.421
##  6 HSPA1A    Linear descending   0.423
##  7 MARCH6    Linear descending   0.445
##  8 MARCH6    Gradient descending 0.461
##  9 HOPX      Linear ascending    0.593
## 10 C3        Linear descending   0.619
## # ... with 237,528 more rows</code></pre>
<p>The resulting data.frame contains the evaluation of all fits for all provided variables. <code><a href="../reference/plotTrajectoryFit.html">plotTrajectoryFit()</a></code> visualizes the fitting results for a particular variable.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># compare the trend of a variable to different models</span>
<span class="fu"><a href="../reference/plotTrajectoryFit.html">plotTrajectoryFit</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>,
                  trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>, 
                  variable <span class="op">=</span> <span class="st">"GFAP"</span>, 
                  display_residuals <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="../reference/legendBottom.html">legendTop</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-trajectory-analysis_files/figure-html/traj-fit-1.png" alt="Figure 4.1. Visulization of the fitting results" width="100%"><p class="caption">
Figure 4.1. Visulization of the fitting results
</p>
</div>
<p>To get a summary of your trajectory assessment use <code><a href="../reference/examineTrajectoryAssessment.html">examineTrajectoryAssessment()</a></code> which displays the distribution of all areas under the curve measured. Use <code><a href="../reference/filterTrends.html">filterTrajectoryTrends()</a></code> or a combination of <code><a href="https://dplyr.tidyverse.org/reference/filter.html">dplyr::filter()</a></code>and <code><a href="https://dplyr.tidyverse.org/reference/pull.html">dplyr::pull()</a></code> on your assessed trajectory data.frame in order to extract the variables following a specific trend to a certain quality-degree.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># example 1: extract all variables that follow the linear descending trend while moving towards the hypoxic area (See Figure 2.1)</span>
<span class="co"># with an auc-evaluation equal to or lower than 2</span>
<span class="va">descending_genes</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/filterTrends.html">filterTrajectoryTrends</a></span><span class="op">(</span>atdf <span class="op">=</span> <span class="va">atdf_genes</span>,
                         limit <span class="op">=</span> <span class="fl">2</span>,
                         trends <span class="op">=</span> <span class="st">"Gradient descending"</span>, 
                         variables_only <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="co"># return a data.frame</span>

<span class="va">descending_genes</span></code></pre></div>
<pre><code>## # A tibble: 29 x 3
## # Groups:   pattern [1]
##    variables pattern               auc
##    &lt;chr&gt;     &lt;chr&gt;               &lt;dbl&gt;
##  1 GFAP      Gradient descending 0.387
##  2 MARCH6    Gradient descending 0.461
##  3 HSPA1A    Gradient descending 0.735
##  4 PDCD6     Gradient descending 0.774
##  5 IL1B      Gradient descending 0.890
##  6 C3        Gradient descending 1.07 
##  7 EFEMP1    Gradient descending 1.28 
##  8 BCAN      Gradient descending 1.33 
##  9 HSPA1B    Gradient descending 1.38 
## 10 BEX1      Gradient descending 1.39 
## # ... with 19 more rows</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># example 2: extract variables that featured a peak along the trajectory </span>
<span class="co"># with auc-evaluation equal to or lower than 4</span>
<span class="va">peaking_genes</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/filterTrends.html">filterTrajectoryTrends</a></span><span class="op">(</span>atdf <span class="op">=</span> <span class="va">atdf_genes</span>,
                         limit <span class="op">=</span> <span class="fl">4</span>,
                         trends <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Early peak"</span>, <span class="st">"Late peak"</span>, <span class="st">"One peak"</span><span class="op">)</span>,
                         variables_only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># return a vector of variables</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">peaking_genes</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "CKB"    "METRN"  "TMSB4X" "LPL"    "CALM1"  "ITGA3"</code></pre>
</div>
<div id="customized-models" class="section level3">
<h3 class="hasAnchor">
<a href="#customized-models" class="anchor"></a>4.2 Customized models</h3>
<p>The number of different trends that can be adopted by the expression of genes and gene-sets along a trajectory is potentially infinite. With the trends introduced in Figure 4.1 we integrated those that struck us a the most common ones to look for. If, however, you are interested in finding the variables that follow a highly specific pattern you can use the trajectory functions that are suffixed with <code>...Customized()</code>. Before modeling spatial trajectories on your own you might want to have a deeper look into the theory and math underlying the concept. <a href="trajectory-analysis-concept.html">Click here</a> to be guided to a more detailed explanation.</p>
<div id="setting-up-a-customized-trajectory-pattern" class="section level4">
<h4 class="hasAnchor">
<a href="#setting-up-a-customized-trajectory-pattern" class="anchor"></a>4.2.1 Setting up a customized trajectory pattern</h4>
<p>Eventually a trajectory plot is composed of the expression averages of binned barcode-spots plotted against the trajectoryâ€™s direction. While the peaking trends might be a bit more complex to generate, the linear ascending trajectory pattern can be emulated quite easily as shown in the example below. What you need in any case to set up a pattern is the number of averages the trajectory of interest is composed of - we refer to that as the trajectory length.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trajectory_length</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getTrajectoryLength.html">getTrajectoryLength</a></span><span class="op">(</span><span class="va">spata_obj</span>, trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>, binwidth <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>

<span class="va">trajectory_length</span></code></pre></div>
<pre><code>## [1] 45</code></pre>
<p>Assuming we`d wanted to fit a linear ascending model we would need to generate a sequence - a numeric vector - of the same length as the trajectory.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">trajectory_direction</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="va">trajectory_length</span>

<span class="va">linear_ascending</span> <span class="op">&lt;-</span> <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org//reference/rescale.html">rescale</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="va">trajectory_length</span>, to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="../reference/plotTrajectoryFeatures.html">plotTrajectoryGenes</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>, genes <span class="op">=</span> <span class="st">"GFAP"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">trajectory_direction</span>, y <span class="op">=</span> <span class="va">linear_ascending</span><span class="op">)</span>,
                     color <span class="op">=</span> <span class="st">"blue"</span>,
                     size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-trajectory-analysis_files/figure-html/get-linear-ascending-pattern-1.png" alt="Figure 4.2 Visualization of a customized linear ascending model" width="100%"><p class="caption">
Figure 4.2 Visualization of a customized linear ascending model
</p>
</div>
<p>While the linear ascending trend isnâ€™t particularly complex any other biologically relevant pattern can be abstracted into a pattern this way and the trend of certain variables (such as the expression of GFAP) along the trajectory can be fitted to that pattern. Apart from constructing customized patterns you can take the actual expression trends of certain variables and take them as the model to screen all the other variables for.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">traj_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getTrajectoryDf.html">getTrajectoryDf</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>,
                           trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>,
                           variables <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"HM_HYPOXIA"</span>, <span class="co"># hypoxia gene set</span>
                                         <span class="st">"METRN"</span><span class="op">)</span>, <span class="co"># METRN gene </span>
                           binwidth <span class="op">=</span> <span class="fl">5</span>, 
                           shift_wider <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">traj_df</span>, <span class="va">trajectory_part</span>, <span class="va">trajectory_order</span>, <span class="va">HM_HYPOXIA</span><span class="op">)</span> </code></pre></div>
<pre><code>## # A tibble: 45 x 3
##    trajectory_part trajectory_order HM_HYPOXIA
##    &lt;chr&gt;                      &lt;int&gt;      &lt;dbl&gt;
##  1 Part 1                         1     0.108 
##  2 Part 1                         2     0.0699
##  3 Part 1                         3     0.0995
##  4 Part 1                         4     0.122 
##  5 Part 1                         5     0.0916
##  6 Part 1                         6     0.148 
##  7 Part 1                         7     0.155 
##  8 Part 1                         8     0.0312
##  9 Part 1                         9     0.101 
## 10 Part 1                        10     0.102 
## # ... with 35 more rows</code></pre>
<p>The rows/observations of a trajectory data.frame obtained with <code><a href="../reference/getTrajectoryDf.html">getTrajectoryDf()</a></code> refer to the bins that are aligned according to the trajectoryâ€™s direction. (The variable <em>trajectory_order</em> corresponds to the direction.)</p>
</div>
<div id="fitting-against-customized-models" class="section level4">
<h4 class="hasAnchor">
<a href="#fitting-against-customized-models" class="anchor"></a>4.2.2 Fitting against customized models</h4>
<p>The variable <em>HM_HYPOXIA</em> of the data.frame <code>traj_df</code> naturally is of the same length as the trajectory itself. Assuming weâ€™d want to screen genes along our defined trajectory to find those that behave similarly to the way <em>HM_HYPOXIA</em> is expressed we can take <em>HM_HYPOXIA</em> as our trend. The argument <code>customized_trends</code>takes either a named list or a data.frame and automatically takes all numeric vectors/variables as models as long as they have the same length as the trajectory.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">atdf_cust</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assessTrajectoryTrends.html">assessTrajectoryTrendsCustomized</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>,
                                           trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>,
                                           customized_trends <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">traj_df</span>, <span class="va">HM_HYPOXIA</span>, <span class="va">METRN</span><span class="op">)</span>, 
                                           variables <span class="op">=</span> <span class="va">all_genes</span>
                                           <span class="op">)</span>

<span class="va">atdf_cust</span></code></pre></div>
<pre><code>## # A tibble: 49,104 x 3
## # Groups:   variables [16,368]
##    variables pattern      auc
##    &lt;chr&gt;     &lt;chr&gt;      &lt;dbl&gt;
##  1 VEGFA     HM_HYPOXIA 0.466
##  2 GAPDH     HM_HYPOXIA 0.536
##  3 LDHA      HM_HYPOXIA 0.537
##  4 LGALS3    HM_HYPOXIA 0.679
##  5 SLC2A3    HM_HYPOXIA 0.745
##  6 MT1X      HM_HYPOXIA 0.756
##  7 IGFBP5    HM_HYPOXIA 0.779
##  8 ENO2      HM_HYPOXIA 0.847
##  9 IRS2      HM_HYPOXIA 0.864
## 10 ADM       HM_HYPOXIA 1.04 
## # ... with 49,094 more rows</code></pre>
<p>Now <em>HM_HYPOXIA</em> is the pattern and as we did with <code>atdf</code> which contained the results from the fitting against predefined models we can use <code><a href="../reference/filterTrends.html">filterTrajectoryTrends()</a></code> to extract the variable names from <code>atdf_cust</code> to get those who resembled the expression dynamics of <em>HM_HYPOXIA</em> to a certain degree.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">similar_to_HM_HYPOXIA</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterTrends.html">filterTrajectoryTrends</a></span><span class="op">(</span><span class="va">atdf_cust</span>, limit <span class="op">=</span> <span class="fl">0.8</span>, trends <span class="op">=</span> <span class="st">"HM_HYPOXIA"</span><span class="op">)</span>

<span class="co"># print names of similar genes</span>
<span class="va">similar_to_HM_HYPOXIA</span></code></pre></div>
<pre><code>## [1] "VEGFA"  "GAPDH"  "LDHA"   "LGALS3" "SLC2A3" "MT1X"   "IGFBP5"</code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plot both in comparison </span>
<span class="fu"><a href="../reference/plotTrajectoryFeatures.html">plotTrajectoryGeneSets</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
                       trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>, 
                       gene_sets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"HM_HYPOXIA"</span><span class="op">)</span>, 
                       display_facets <span class="op">=</span> <span class="cn">TRUE</span>, 
                       clrp <span class="op">=</span> <span class="st">"default"</span><span class="op">)</span>

<span class="fu"><a href="../reference/plotTrajectoryFeatures.html">plotTrajectoryGenes</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
                    trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>, 
                    genes <span class="op">=</span> <span class="va">similar_to_HM_HYPOXIA</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>, 
                    display_facets <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-trajectory-analysis_files/figure-html/filterTrajectoryTrends()-on-atdf_cust-1.png" alt="Figure 4.3 Comparison of the screening results against the HM_HYPOXIA-model" width="50%"><img src="spata-v2-trajectory-analysis_files/figure-html/filterTrajectoryTrends()-on-atdf_cust-2.png" alt="Figure 4.3 Comparison of the screening results against the HM_HYPOXIA-model" width="50%"><p class="caption">
Figure 4.3 Comparison of the screening results against the HM_HYPOXIA-model
</p>
</div>
<p>The same can be done for gene sets similar to the expression of METRN.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">similar_to_METRN</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterTrends.html">filterTrajectoryTrends</a></span><span class="op">(</span><span class="va">atdf_cust</span>, limit <span class="op">=</span> <span class="fl">2</span>, trends <span class="op">=</span> <span class="st">"METRN"</span><span class="op">)</span>

<span class="co"># print names of similar gene sets</span>
<span class="va">similar_to_METRN</span></code></pre></div>
<pre><code>## [1] "NKAIN4" "CHKA"   "EDNRB"  "HIPK2"</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plot both in comparison </span>
<span class="fu"><a href="../reference/plotTrajectoryFeatures.html">plotTrajectoryGenes</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
                    trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>, 
                    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"METRN"</span><span class="op">)</span>, 
                    display_facets <span class="op">=</span> <span class="cn">TRUE</span>, 
                    clrp <span class="op">=</span> <span class="st">"default"</span><span class="op">)</span>

<span class="fu"><a href="../reference/plotTrajectoryFeatures.html">plotTrajectoryGenes</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
                    trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>, 
                    genes <span class="op">=</span> <span class="va">similar_to_METRN</span>, 
                    display_facets <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-trajectory-analysis_files/figure-html/filterTrajectoryTrends()-on-atdf_c2-1.png" alt="Figure 4.4 Comparison of the screening results against the METRN-model" width="50%"><img src="spata-v2-trajectory-analysis_files/figure-html/filterTrajectoryTrends()-on-atdf_c2-2.png" alt="Figure 4.4 Comparison of the screening results against the METRN-model" width="50%"><p class="caption">
Figure 4.4 Comparison of the screening results against the METRN-model
</p>
</div>
</div>
</div>
</div>
<div id="display-trajectory-trends-with-heatmaps" class="section level2">
<h2 class="hasAnchor">
<a href="#display-trajectory-trends-with-heatmaps" class="anchor"></a>5. Display trajectory trends with heatmaps</h2>
<p>Displaying the expression trends of too many variables in a lineplot gets unclear quite rapidly. In order to deal with several variables use <code><a href="../reference/plotTrajectoryHeatmap.html">plotTrajectoryHeatmap()</a></code>. While the trajectoryâ€™s directory is still displayed from left to right the dynamic of the gene-setsâ€™ expression values is now displayed by the gradual change of a rowâ€™s color instead of a lineâ€™s slope in a lineplot. The advantage of using heatmaps to display dynamic expression changes along trajectories becomes obvious when the number of variables to display rises.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">descending_genes_vec</span> <span class="op">&lt;-</span> <span class="va">descending_genes</span><span class="op">$</span><span class="va">variables</span>

<span class="va">hm_colors</span> <span class="op">&lt;-</span> <span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/viridisLite/man/viridis.html">inferno</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>

<span class="fu"><a href="../reference/plotTrajectoryHeatmap.html">plotTrajectoryHeatmap</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
                      trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>,
                      variables <span class="op">=</span> <span class="va">descending_genes_vec</span>,
                      arrange_rows <span class="op">=</span> <span class="st">"maxima"</span>,
                      colors <span class="op">=</span> <span class="va">hm_colors</span>,
                      show_rownames <span class="op">=</span> <span class="cn">TRUE</span>,
                      split_columns <span class="op">=</span> <span class="cn">FALSE</span>, 
                      smooth_span <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-trajectory-analysis_files/figure-html/trajectory-heatmap-1-1.png" alt="Figure 5.1 A heatmap displaying linear descending trends" width="100%"><p class="caption">
Figure 5.1 A heatmap displaying linear descending trends
</p>
</div>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTrajectoryHeatmap.html">plotTrajectoryHeatmap</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
                      trajectory_name <span class="op">=</span> <span class="st">"tumor-areas"</span>,
                      variables <span class="op">=</span> <span class="va">peaking_genes</span>, 
                      arrange_rows <span class="op">=</span> <span class="st">"maxima"</span>,
                      colors <span class="op">=</span> <span class="va">hm_colors</span>,
                      show_row_names <span class="op">=</span> <span class="cn">FALSE</span>, 
                      split_columns <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># splits the heatmap to highlight the trajectory parts</span>
                      smooth_span <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-trajectory-analysis_files/figure-html/trajectory-heatmap-2-1.png" alt="Figure 5.2 A heatmap displaying the gradual change of peaking genes" width="100%"><p class="caption">
Figure 5.2 A heatmap displaying the gradual change of peaking genes
</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Jan Kueckelhaus, Dieter-Henrik Heiland.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
