<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Segmentation &amp; Clustering • SPATA2</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Segmentation &amp; Clustering">
<meta property="og:description" content="SPATA2">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SPATA2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/spata-v2-introduction-and-philosophy.html">About</a>
</li>
<li>
  <a href="../articles/spata-v2-installation.html">Installation</a>
</li>
<li>
  <a href="../articles/spata-v2-terminology.html">Terminology</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Around the Spata-Object</li>
    <li>
      <a href="../articles/spata-v2-object-initiation-and-manipulation.html">Object Set Up and Manipulation</a>
    </li>
    <li>
      <a href="../articles/spata-v2-extract-add-and-join-data.html">Extracting, Adding and Joining Data</a>
    </li>
    <li>
      <a href="../articles/spata-v2-platform-compatibility.html">Integrating Different Platforms</a>
    </li>
    <li class="dropdown-header">Analysis &amp; Computation</li>
    <li>
      <a href="../articles/spata-v2-autoencoder.html">Autoencoder Denoising</a>
    </li>
    <li>
      <a href="../articles/spata-v2-cnv-analysis.html">Copy Number Variations</a>
    </li>
    <li>
      <a href="../articles/spata-v2-de-analysis.html">Differential Expression Analysis</a>
    </li>
    <li>
      <a href="../articles/spata-v2-segmentation-and-clustering.html">Spatial Segmentation &amp; Clustering</a>
    </li>
    <li>
      <a href="../articles/spata-v2-trajectory-analysis.html">Spatial Trajectories</a>
    </li>
    <li class="dropdown-header">Visualization</li>
    <li>
      <a href="../articles/spata-v2-plotting-surface.html">Surface Plotting</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-dimensional-reduction.html">Dimensional Reduction</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-descriptive-statistics.html">Descriptive Statistics</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-miscellaneous.html">Miscellaneous</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="spata-v2-segmentation-and-clustering_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Segmentation &amp; Clustering</h1>
            
      
      
      <div class="hidden name"><code>spata-v2-segmentation-and-clustering.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>1. Introduction</h1>
<p>Discrete (syn. categorical) features divide a sample’s barcode-spots into experimental groups whoose properties can be compared against each other. The grouping of barcode-spots can be the result of clustering algorithms or manual spatial segmentation. This tutorial shows how to do both in <strong>SPATA2</strong>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://themilolab.com/">SPATA2</a></span><span class="op">)</span>

<span class="va">spata_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/loadSpataObject.html">loadSpataObject</a></span><span class="op">(</span><span class="st">"data/spata-obj-gbm275.RDS"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-segmentation-and-clustering_files/figure-html/cluster-plot-1.png" alt="Figure 1.1 Grouping of barcode-spots by clusters" width="100%"><p class="caption">
Figure 1.1 Grouping of barcode-spots by clusters
</p>
</div>
<div class="figure">
<img src="spata-v2-segmentation-and-clustering_files/figure-html/segm-plot-1.png" alt="Figure 1.2 Grouping of barcode-spots by manual, spatial segmentation" width="100%"><p class="caption">
Figure 1.2 Grouping of barcode-spots by manual, spatial segmentation
</p>
</div>
</div>
<div id="spatial-segmentation" class="section level1">
<h1 class="hasAnchor">
<a href="#spatial-segmentation" class="anchor"></a>2. Spatial segmentation</h1>
<p>The spatial dimension of spatial transcriptomics invites to compare regions of interest against each other with respect to their gene expression or other features. The segment belonging of every barcode-spot is stored as an additional variable in the object’s feature data. Assuming that you have not assigned any barcode-spots to segments so far your feature data.frame will look like this:</p>
<pre><code>## # A tibble: 3,734 x 3
##    barcodes           sample segmentation
##    &lt;chr&gt;              &lt;chr&gt;  &lt;chr&gt;       
##  1 AAACAAGTATCTCCCA-1 T275   none        
##  2 AAACACCAATAACTGC-1 T275   none        
##  3 AAACAGAGCGACTCCT-1 T275   none        
##  4 AAACAGCTTTCAGAAG-1 T275   none        
##  5 AAACAGGGTCTATATT-1 T275   none        
##  6 AAACAGTGTTCCTGGG-1 T275   none        
##  7 AAACATTTCCCGGATT-1 T275   none        
##  8 AAACCGGGTAGGTACC-1 T275   none        
##  9 AAACCGTTCGTCCAGG-1 T275   none        
## 10 AAACCTAAGCAGCCGG-1 T275   none        
## # ... with 3,724 more rows</code></pre>
<p>Barcode spots that have not been assigned to a segment are denoted with <em>‘none’</em> as there isn’t a segment to which they belong.</p>
<p>In order to to assign barcode-spots to a segment make use of the <code><a href="../reference/createSegmentation.html">createSegmentation()</a></code>-function. It opens am interactive application in which you can determine the extent of every segment by determining the vertices of the polygon which in turn determines the segment’s borders.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">spata_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createSegmentation.html">createSegmentation</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<br><div class="figure">
<img src="images/create-segmentation-fig-1.1.png" alt="Figure 2.1 Easy interactive segmentation" width="100%"><p class="caption">
Figure 2.1 Easy interactive segmentation
</p>
</div>
<br><div class="figure">
<img src="images/create-segmentation-fig-1.2.png" alt="Figure 2.2 Highlight and save the segment" width="100%"><p class="caption">
Figure 2.2 Highlight and save the segment
</p>
</div>
<br><div class="figure">
<img src="images/create-segmentation-fig-1.3.png" alt="Figure 2.3 Save the segment and continue with the next area of interest" width="100%"><p class="caption">
Figure 2.3 Save the segment and continue with the next area of interest
</p>
</div>
<div class="figure">
<img src="images/create-segmentation-fig-1.4.png" alt="Figure 2.4 Set the transparency to max in order to see the histology" width="100%"><p class="caption">
Figure 2.4 Set the transparency to max in order to see the histology
</p>
</div>
<p>In order to visualize the current segmentation of your sample run <code><a href="../reference/plotSegmentation.html">plotSegmentation()</a></code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># display the current segmentation</span>
<span class="fu"><a href="../reference/plotSegmentation.html">plotSegmentation</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, pt_size <span class="op">=</span> <span class="fl">1.9</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-segmentation-and-clustering_files/figure-html/plot-segmentation-1.png" alt="Figure 2.5 The current segmentation" width="100%"><p class="caption">
Figure 2.5 The current segmentation
</p>
</div>
<p>The segment belonging is now a usable, discrete variable of your spata-object’s feature data - a discrete feature - , which can be used in <strong>SPATA2</strong> as any other discrete feature and extracted in a tidy-data fashion as well.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getFeatureVariables.html">getFeatureVariables</a></span><span class="op">(</span><span class="va">spata_obj</span>, features <span class="op">=</span> <span class="st">"segmentation"</span>, return <span class="op">=</span> <span class="st">"data.frame"</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 3,734 x 3
##    barcodes           sample segmentation    
##    &lt;chr&gt;              &lt;chr&gt;  &lt;chr&gt;           
##  1 AAACAAGTATCTCCCA-1 T275   none            
##  2 AAACACCAATAACTGC-1 T275   oxid-phosph-high
##  3 AAACAGAGCGACTCCT-1 T275   none            
##  4 AAACAGCTTTCAGAAG-1 T275   none            
##  5 AAACAGGGTCTATATT-1 T275   none            
##  6 AAACAGTGTTCCTGGG-1 T275   oxid-phosph-high
##  7 AAACATTTCCCGGATT-1 T275   none            
##  8 AAACCGGGTAGGTACC-1 T275   none            
##  9 AAACCGTTCGTCCAGG-1 T275   oxid-phosph-high
## 10 AAACCTAAGCAGCCGG-1 T275   none            
## # ... with 3,724 more rows</code></pre>
<p>If you need the barcodes of a segment’s barcode spots you can obtain them for example via <code><a href="../reference/getCoordsDf.html">getSegmentDf()</a></code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getCoordsDf.html">getSegmentDf</a></span><span class="op">(</span><span class="va">spata_obj</span>, segment_names <span class="op">=</span> <span class="st">"oxid-phosph-high"</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 632 x 5
##    barcodes           sample     x     y segmentation    
##    &lt;chr&gt;              &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;           
##  1 AAACACCAATAACTGC-1 T275   132.   429. oxid-phosph-high
##  2 AAACAGTGTTCCTGGG-1 T275   219.   518. oxid-phosph-high
##  3 AAACCGTTCGTCCAGG-1 T275   216.   385. oxid-phosph-high
##  4 AAACGAAGATGGAGTA-1 T275    77.2  422. oxid-phosph-high
##  5 AAACTTAATTGCACGC-1 T275   106.   460. oxid-phosph-high
##  6 AAAGGCCCTATAATAC-1 T275   142.   473. oxid-phosph-high
##  7 AAAGTAGCATTGCTCA-1 T275   161.   378. oxid-phosph-high
##  8 AAATCGCGGAAGGAGT-1 T275    80.9  416. oxid-phosph-high
##  9 AAATGGCCCGTGCCCT-1 T275   125.   365. oxid-phosph-high
## 10 AAATTGATAGTCCTTT-1 T275   176.   404. oxid-phosph-high
## # ... with 622 more rows</code></pre>
</div>
<div id="clustering" class="section level1">
<h1 class="hasAnchor">
<a href="#clustering" class="anchor"></a>3. Clustering</h1>
<p>In transcriptomic analysis clustering divides barcode-spots into groups based on the similarity of their gene-expression profiles. There are several algorithms out there that can be used to divide your sample into subgroups. While <code><a href="../reference/initiateSpataObject_10X.html">initiateSpataObject_10X()</a></code> integrates the clustering algorithm used by the <strong>Seurat</strong>-package there are actually many more. As the use of clustering is highly depending on the biological question it makes sense to use several approaches and algorithms. <strong>SPATA2</strong> currently offers the functions <code><a href="../reference/findSeuratClusters.html">findSeuratClusters()</a></code>, <code><a href="../reference/findMonocleClusters.html">findMonocleClusters()</a></code> and <code><a href="../reference/findNearestNeighbourClusters.html">findNearestNeighbourClusters()</a></code> which are wrapper around widely used clustering algorithms. The workflow of all functions is equal: Run the algorithm, obtain the results in a data.frame and add the clustering variables of interest to your spata-object via <code><a href="../reference/addFeatures.html">addFeatures()</a></code>. The following code chunk exemplifies that using <code><a href="../reference/findMonocleClusters.html">findMonocleClusters()</a></code>.</p>
<p>The package monocle3 makes use of louvain- and leidenbase clustering with the <code><a href="https://rdrr.io/pkg/monocle3/man/cluster_cells.html">cluster_cells()</a></code>-function. This function takes several additional parameters that can be used to tweak the clustering-algorithm’s performance with respect to the properties of every individual sample. <strong>SPATA2</strong>’s <code><a href="../reference/findMonocleClusters.html">findMonocleClusters()</a></code> is a wrapper around monocle3’s clustering options. It iterates over the parameters you provide and returns a tidy spata data.frame containing each result as a separate variable.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">monocle_clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findMonocleClusters.html">findMonocleClusters</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
                                        preprocess_method <span class="op">=</span> <span class="st">"PCA"</span>, 
                                        reduction_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"UMAP"</span>, <span class="st">"PCA"</span>, <span class="st">"tSNE"</span><span class="op">)</span>, 
                                        cluster_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"leiden"</span>, <span class="st">"louvain"</span><span class="op">)</span>, 
                                        k <span class="op">=</span> <span class="fl">5</span>, 
                                        num_iter <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>

<span class="co"># output</span>
<span class="va">monocle_clusters</span></code></pre></div>
<pre><code>## # A tibble: 3,734 x 7
##    barcodes  cluster_leiden_~ cluster_louvain~ cluster_leiden_~ cluster_louvain~
##    &lt;chr&gt;     &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;           
##  1 AAACAAGT~ Cluster 1        Cluster 29       Cluster 1        Cluster 2       
##  2 AAACACCA~ Cluster 5        Cluster 43       Cluster 1        Cluster 6       
##  3 AAACAGAG~ Cluster 3        Cluster 16       Cluster 1        Cluster 1       
##  4 AAACAGCT~ Cluster 2        Cluster 27       Cluster 1        Cluster 5       
##  5 AAACAGGG~ Cluster 5        Cluster 37       Cluster 1        Cluster 3       
##  6 AAACAGTG~ Cluster 4        Cluster 14       Cluster 1        Cluster 2       
##  7 AAACATTT~ Cluster 4        Cluster 36       Cluster 1        Cluster 2       
##  8 AAACCGGG~ Cluster 2        Cluster 38       Cluster 1        Cluster 5       
##  9 AAACCGTT~ Cluster 1        Cluster 17       Cluster 1        Cluster 2       
## 10 AAACCTAA~ Cluster 4        Cluster 8        Cluster 1        Cluster 6       
## # ... with 3,724 more rows, and 2 more variables: cluster_leiden_tSNE_k5 &lt;chr&gt;,
## #   cluster_louvain_tSNE_k5 &lt;chr&gt;</code></pre>
<p>Use the helper function <code><a href="../reference/examineClusterResults.html">examineClusterResults()</a></code> to quickly see all unique cluster names and to evaluate which clustering results come into question.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># output</span>
<span class="fu"><a href="../reference/examineClusterResults.html">examineClusterResults</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">monocle_clusters</span><span class="op">)</span></code></pre></div>
<pre><code>## $cluster_leiden_UMAP_k5
## [1] "Cluster 1" "Cluster 5" "Cluster 3" "Cluster 2" "Cluster 4" "Cluster 6"
## [7] "Cluster 7"
## 
## $cluster_louvain_UMAP_k5
##  [1] "Cluster 29" "Cluster 43" "Cluster 16" "Cluster 27" "Cluster 37"
##  [6] "Cluster 14" "Cluster 36" "Cluster 38" "Cluster 17" "Cluster 8" 
## [11] "Cluster 24" "Cluster 23" "Cluster 45" "Cluster 44" "Cluster 7" 
## [16] "Cluster 40" "Cluster 34" "Cluster 1"  "Cluster 32" "Cluster 11"
## [21] "Cluster 21" "Cluster 6"  "Cluster 13" "Cluster 42" "Cluster 31"
## [26] "Cluster 41" "Cluster 12" "Cluster 20" "Cluster 26" "Cluster 2" 
## [31] "Cluster 4"  "Cluster 28" "Cluster 25" "Cluster 18" "Cluster 46"
## [36] "Cluster 39" "Cluster 30" "Cluster 33" "Cluster 35" "Cluster 10"
## [41] "Cluster 15" "Cluster 9"  "Cluster 3"  "Cluster 19" "Cluster 5" 
## [46] "Cluster 22"
## 
## $cluster_leiden_PCA_k5
## [1] "Cluster 1" "Cluster 2"
## 
## $cluster_louvain_PCA_k5
## [1] "Cluster 2" "Cluster 6" "Cluster 1" "Cluster 5" "Cluster 3" "Cluster 8"
## [7] "Cluster 9" "Cluster 7" "Cluster 4"
## 
## $cluster_leiden_tSNE_k5
## [1] "Cluster 5" "Cluster 3" "Cluster 2" "Cluster 4" "Cluster 1" "Cluster 6"
## [7] "Cluster 7" "Cluster 8"
## 
## $cluster_louvain_tSNE_k5
##  [1] "Cluster 10" "Cluster 28" "Cluster 6"  "Cluster 22" "Cluster 35"
##  [6] "Cluster 48" "Cluster 44" "Cluster 41" "Cluster 17" "Cluster 24"
## [11] "Cluster 25" "Cluster 12" "Cluster 23" "Cluster 39" "Cluster 19"
## [16] "Cluster 29" "Cluster 27" "Cluster 11" "Cluster 4"  "Cluster 33"
## [21] "Cluster 30" "Cluster 43" "Cluster 37" "Cluster 49" "Cluster 16"
## [26] "Cluster 40" "Cluster 21" "Cluster 13" "Cluster 3"  "Cluster 26"
## [31] "Cluster 46" "Cluster 2"  "Cluster 32" "Cluster 14" "Cluster 7" 
## [36] "Cluster 38" "Cluster 47" "Cluster 9"  "Cluster 31" "Cluster 42"
## [41] "Cluster 15" "Cluster 20" "Cluster 18" "Cluster 8"  "Cluster 34"
## [46] "Cluster 5"  "Cluster 36" "Cluster 45" "Cluster 1"</code></pre>
<p>It seems as if the results <em>‘cluster_leiden_UMAP_k5’</em>, <em>‘cluster_leiden_tSNE_k5’</em>, <em>’cluster_louvain_PCA_k5</em>’ are the only ones that make sense to work with as the other three either result in way to many or way to less groups.</p>
<p>Via <code><a href="../reference/addFeatures.html">addFeatures()</a></code> you can add all clustering results to your spata-object simultaneously, which makes them available for every <strong>SPATA2</strong>-function. Each variable in <code>monocle_clusters</code> represents a possible option to assign barcodes to experimental groups. And since they are stored as individual variables they can be analyzed and visualized one by one.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># feature names before adding</span>
<span class="fu"><a href="../reference/getFeatureNames.html">getFeatureNames</a></span><span class="op">(</span><span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code>##            numeric            integer            numeric            numeric 
##   "nCount_Spatial" "nFeature_Spatial"       "percent.mt"       "percent.RB" 
##             factor          character             factor             factor 
##  "seurat_clusters"     "segmentation"      "Leiden_UMAP"          "cluster" 
##            logical 
##             "keep"</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># add the cluster results</span>
<span class="va">spata_obj</span> <span class="op">&lt;-</span> 
  <span class="fu"><a href="../reference/addFeatures.html">addFeatures</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
              feature_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"cluster_leiden_UMAP_k5"</span>, <span class="st">"cluster_leiden_tSNE_k5"</span>,<span class="st">"cluster_louvain_PCA_k5"</span><span class="op">)</span>, 
              feature_df <span class="op">=</span> <span class="va">monocle_clusters</span>,
              overwrite <span class="op">=</span> <span class="cn">TRUE</span>,
              key <span class="op">=</span> <span class="st">"barcodes"</span><span class="op">)</span>

<span class="co"># feature names afterwards</span>
<span class="fu"><a href="../reference/getFeatureNames.html">getFeatureNames</a></span><span class="op">(</span><span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code>##                  numeric                  integer                  numeric 
##         "nCount_Spatial"       "nFeature_Spatial"             "percent.mt" 
##                  numeric                   factor                character 
##             "percent.RB"        "seurat_clusters"           "segmentation" 
##                   factor                   factor                  logical 
##            "Leiden_UMAP"                "cluster"                   "keep" 
##                character                character                character 
## "cluster_leiden_UMAP_k5" "cluster_leiden_tSNE_k5" "cluster_louvain_PCA_k5"</code></pre>
<p>Continue by visualizing your results or by investigating their properties (e.g. <a href="spata-v2-de-analysis.html">differentially expressed genes</a>).</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotSurface.html">plotSurface</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>,
            color_by <span class="op">=</span> <span class="st">"cluster_leiden_UMAP_k5"</span>,
            pt_size <span class="op">=</span> <span class="fl">1.9</span>,
            pt_clrp <span class="op">=</span> <span class="st">"jama"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"Leiden UMAP"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-segmentation-and-clustering_files/figure-html/plot-clusters-1.png" alt="Figure 3.1 Easy visualization of different clustering results - Leiden UMAP" width="100%"><p class="caption">
Figure 3.1 Easy visualization of different clustering results - Leiden UMAP
</p>
</div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotSurface.html">plotSurface</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>,
            color_by <span class="op">=</span> <span class="st">"cluster_leiden_tSNE_k5"</span>,
            pt_size <span class="op">=</span> <span class="fl">1.9</span>,
            pt_clrp <span class="op">=</span> <span class="st">"jco"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"Leiden tSNE"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-segmentation-and-clustering_files/figure-html/plot-clusters-2-1.png" alt="Figure 3.2 Easy visualization of different clustering results - Leiden tSNE" width="100%"><p class="caption">
Figure 3.2 Easy visualization of different clustering results - Leiden tSNE
</p>
</div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotSurface.html">plotSurface</a></span><span class="op">(</span><span class="va">spata_obj</span>,
            color_by <span class="op">=</span> <span class="st">"cluster_louvain_PCA_k5"</span>,
            pt_size <span class="op">=</span> <span class="fl">1.9</span>,
            pt_clrp <span class="op">=</span> <span class="st">"nejm"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"Louvain PCA"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-segmentation-and-clustering_files/figure-html/plot-clusters-3-1.png" alt="Figure 3.3 Easy visualization of different clustering results - Louvain PCA" width="100%"><p class="caption">
Figure 3.3 Easy visualization of different clustering results - Louvain PCA
</p>
</div>
<p>Remember that you can always rename clustering variables with <code><a href="../reference/renameFeatures.html">renameFeatures()</a></code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># rename the clustering variable </span>
<span class="va">spata_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/renameFeatures.html">renameFeatures</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, <span class="st">"Leiden_UMAP"</span> <span class="op">=</span> <span class="st">"cluster_leiden_UMAP_k5"</span><span class="op">)</span>

<span class="co"># print features names</span>
<span class="fu"><a href="../reference/getFeatureNames.html">getFeatureNames</a></span><span class="op">(</span><span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code>##                  numeric                  integer                  numeric 
##         "nCount_Spatial"       "nFeature_Spatial"             "percent.mt" 
##                  numeric                   factor                character 
##             "percent.RB"        "seurat_clusters"           "segmentation" 
##                   factor                   factor                  logical 
##            "Leiden_UMAP"                "cluster"                   "keep" 
##                character                character 
## "cluster_leiden_tSNE_k5" "cluster_louvain_PCA_k5"</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Jan Kueckelhaus, Dieter-Henrik Heiland.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
