<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>SPATA Terminology • SPATA2</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="SPATA Terminology">
<meta property="og:description" content="SPATA2">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SPATA2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/spata-v2-introduction-and-philosophy.html">About</a>
</li>
<li>
  <a href="../articles/spata-v2-installation.html">Installation</a>
</li>
<li>
  <a href="../articles/spata-v2-terminology.html">Terminology</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Around the Spata-Object</li>
    <li>
      <a href="../articles/spata-v2-object-initiation-and-manipulation.html">Object Set Up and Manipulation</a>
    </li>
    <li>
      <a href="../articles/spata-v2-extract-add-and-join-data.html">Extracting, Adding and Joining Data</a>
    </li>
    <li>
      <a href="../articles/spata-v2-platform-compatibility.html">Integrating Different Platforms</a>
    </li>
    <li class="dropdown-header">Analysis &amp; Computation</li>
    <li>
      <a href="../articles/spata-v2-autoencoder.html">Autoencoder Denoising</a>
    </li>
    <li>
      <a href="../articles/spata-v2-cnv-analysis.html">Copy Number Variations</a>
    </li>
    <li>
      <a href="../articles/spata-v2-de-analysis.html">Differential Expression Analysis</a>
    </li>
    <li>
      <a href="../articles/spata-v2-segmentation-and-clustering.html">Spatial Segmentation &amp; Clustering</a>
    </li>
    <li>
      <a href="../articles/spata-v2-trajectory-analysis.html">Spatial Trajectories</a>
    </li>
    <li class="dropdown-header">Visualization</li>
    <li>
      <a href="../articles/spata-v2-plotting-surface.html">Surface Plotting</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-dimensional-reduction.html">Dimensional Reduction</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-descriptive-statistics.html">Descriptive Statistics</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-miscellaneous.html">Miscellaneous</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="spata-v2-terminology_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>SPATA Terminology</h1>
            
      
      
      <div class="hidden name"><code>spata-v2-terminology.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>1. Introduction</h2>
<p>In order to make working with the <strong>SPATA2</strong>*-package as easy and as intuitive as possible it’s functions are divided into families with respect to what they do. Their main functionality is described by the first verb their name is composed of which means that you can easily skim through all of them by typing <code>SPATA::</code> into your console.</p>
</div>
<div id="functions" class="section level2">
<h2 class="hasAnchor">
<a href="#functions" class="anchor"></a>2. Functions</h2>
<ol style="list-style-type: decimal">
<li>
<strong>create</strong>-functions make you create new content and then add it immediately to your spata-object. They return an updated version of your spata-object. (e.g. <code><a href="../reference/createSegmentation.html">createSegmentation()</a></code>, <code><a href="../reference/createTrajectories.html">createTrajectories()</a></code>)</li>
<li>
<strong>get</strong>-functions extract data from your spata-object in a tidy data fashion as explained above in order to allow for personal analysis beyond what <strong>SPATA2</strong>* currently offers (e.g. <code><a href="../reference/getCoordsDf.html">getCoordsDf()</a></code>, <code><a href="../reference/getGroupingOptions.html">getGroupingOptions()</a></code>, <code><a href="../reference/getFeatureNames.html">getFeatureNames()</a></code>)</li>
<li>
<strong>joinWith</strong>-functions take extracted data.frames and join information to them via the <em>barcodes</em>-variable. They return data.frames. (e.g. <code><a href="../reference/joinWith.html">joinWithGeneSets()</a></code>)</li>
<li>
<strong>add</strong>/<strong>discard</strong>-functions add or discard content (e.g. from your personal analysis) to your spata-object via key-variables to make it available for all <strong>SPATA2</strong>*-intern functions. They return an updated version of your spata-object. (e.g. <code><a href="../reference/addFeatures.html">addFeatures()</a></code>, <code><a href="../reference/addGeneSet.html">addGeneSet()</a></code>, <code><a href="../reference/discardGenes.html">discardGenes()</a></code>)</li>
<li>
<strong>set</strong>-functions overwrite the content they refer to and are used for slots that do not allow more than one thing to occupy them. They return an updated version of your spata-object. (e.g. there can be several expression matrices that can bee added with <code><a href="../reference/addExpressionMatrix.html">addExpressionMatrix()</a></code> but there can be only one active expression matrix that can be denoted with <code><a href="../reference/setActiveExpressionMatrix.html">setActiveExpressionMatrix()</a></code>.)</li>
<li>
<strong>plot</strong>-functions plot content. They return - in almost all cases - a heatmap from the <code>pheatmap</code>-package or a ggplot-object that can be further customized according to the rules of the ggplot2-framework. (e.g. <code><a href="../reference/plotSurface.html">plotSurface()</a></code>)</li>
<li>&amp; 8. <strong>find</strong>/<strong>assess</strong>-functions generally perform clustering, screening or modelling tasks. The results are <strong>not</strong> stored in the spata-object right away! Therefore they return data.frames. You can use <strong>examine</strong>-functions to examine what has been found and then you can decide how to proceed. (e.g. <code><a href="../reference/findMonocleClusters.html">findMonocleClusters()</a></code>, <code><a href="../reference/assessTrajectoryTrends.html">assessTrajectoryTrends()</a></code>)</li>
<li>
<strong>initiate</strong>-functions create spata-objects from scratch by relying on different input types. They return the spata-object you will work with throughout this platform.</li>
<li>
<strong>transform</strong>-functions assure compatibility between different platforms by transforming the objects into each other. (e.g. <code><a href="../reference/transformSpataToCDS.html">transformSpataToCDS()</a></code> for <code>monocle3</code>)</li>
<li>
<strong>adjust</strong>/<strong>rename</strong>-functions adjust the content of the slot they refer to without necessarily overwriting it completely. They return an updated version of your spata-object. (e.g. <code><a href="../reference/adjustDefaultInstructions.html">adjustDirectoryInstructions()</a></code>, <code><a href="../reference/adjustDefaultInstructions.html">adjustDefaultInstructions()</a></code>, <code><a href="../reference/renameFeatures.html">renameFeatures()</a></code>, <code><a href="../reference/renameGroups.html">renameGroups()</a></code>)</li>
<li>
<strong>valid</strong>-functions simply return valid input options for a variety of arguments. (e.g.<code><a href="../reference/validActivationFunctions.html">validColorPalettes()</a></code>, <code><a href="../reference/validActivationFunctions.html">validActivationFunctions()</a></code>)</li>
<li>
<strong>run</strong>/<strong>compute</strong>-functions cover computationally more intensive topics and store the results in the spata-object. They return an updated version of your spata-object. (e.g. <code><a href="../reference/computeGeneMetaData.html">computeGeneMetaData()</a></code>, <code>runDeAnalsis()</code>, <code><a href="../reference/runAutoencoderDenoising.html">runAutoencoderDenoising()</a></code>)</li>
</ol>
</div>
<div id="arguments" class="section level2">
<h2 class="hasAnchor">
<a href="#arguments" class="anchor"></a>3. Arguments</h2>
<p>As every function accomplishes something unique every function has it’s own special arguments. Still, we attempted to create argument-families that usually refer to the same aspects.</p>
<div id="spata-object" class="section level4">
<h4 class="hasAnchor">
<a href="#spata-object" class="anchor"></a>1. SPATA-object</h4>
<ol style="list-style-type: decimal">
<li>
<code>object</code> takes your spata-object from which it gets all the information. It is the most frequent argument used and usually the first one of every function that needs it.</li>
</ol>
</div>
<div id="data-input" class="section level4">
<h4 class="hasAnchor">
<a href="#data-input" class="anchor"></a>2. Data input</h4>
<p>As <strong>SPATA2</strong>* draws it’s main power from the tidyverse data.frames are a vital part of the communication between functions and of the framework we attempt to provide. Whenever a function takes a data.frame as an input it will contain an argument of the <code>df</code>-family.</p>
<ol style="list-style-type: decimal">
<li>
<code>spata_df</code> refers to data.frames that contain at least the variables <em>barcodes</em> and <em>sample</em> (obtained by e.g. <code><a href="../reference/getSpataDf.html">getSpataDf()</a></code>*).</li>
</ol>
<pre><code>## # A tibble: 3,734 x 2
##    barcodes           sample
##    &lt;chr&gt;              &lt;chr&gt; 
##  1 AAACAAGTATCTCCCA-1 T275  
##  2 AAACACCAATAACTGC-1 T275  
##  3 AAACAGAGCGACTCCT-1 T275  
##  4 AAACAGCTTTCAGAAG-1 T275  
##  5 AAACAGGGTCTATATT-1 T275  
##  6 AAACAGTGTTCCTGGG-1 T275  
##  7 AAACATTTCCCGGATT-1 T275  
##  8 AAACCGGGTAGGTACC-1 T275  
##  9 AAACCGTTCGTCCAGG-1 T275  
## 10 AAACCTAAGCAGCCGG-1 T275  
## # ... with 3,724 more rows</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>
<code>coords_df</code> refers to <em>coordinate data.frames</em> which contain at least the variables <em>barcodes</em>, <em>sample</em>, <em>x</em> and <em>y</em> (obtained by e.g.<code><a href="../reference/getCoordsDf.html">getCoordsDf()</a></code>*).</li>
</ol>
<pre><code>## # A tibble: 3,734 x 4
##    barcodes           sample     x     y
##    &lt;chr&gt;              &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;
##  1 AAACAAGTATCTCCCA-1 T275    435.  374.
##  2 AAACACCAATAACTGC-1 T275    132.  429.
##  3 AAACAGAGCGACTCCT-1 T275    407.  145.
##  4 AAACAGCTTTCAGAAG-1 T275     96   327 
##  5 AAACAGGGTCTATATT-1 T275    110.  353.
##  6 AAACAGTGTTCCTGGG-1 T275    219.  518.
##  7 AAACATTTCCCGGATT-1 T275    416.  443.
##  8 AAACCGGGTAGGTACC-1 T275    165.  321.
##  9 AAACCGTTCGTCCAGG-1 T275    216.  385.
## 10 AAACCTAAGCAGCCGG-1 T275    365.  468.
## # ... with 3,724 more rows</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>
<code>dea_df</code> refers to <em>differential expression data.frames</em> which contain the results of differential gene expression analysis. (obtained by e.g. <code><a href="../reference/getDeaResultsDf.html">getDeaResultsDf()</a></code>).</li>
</ol>
<pre><code>## # A tibble: 6,441 x 7
##       p_val avg_logFC pct.1 pct.2 p_val_adj seurat_clusters gene   
##       &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;           &lt;chr&gt;  
##  1 1.20e- 3      6.68 0.965 0.96   1.00e+ 0 0               DPYSL2 
##  2 3.27e- 5      4.66 0.984 0.988  7.05e- 1 0               AQP4   
##  3 1.31e-40      3.57 0.909 0.8    2.82e-36 0               VOPP1  
##  4 2.01e- 9      3.44 0.842 0.883  4.34e- 5 0               PON2   
##  5 7.85e- 5      2.88 0.721 0.76   1.00e+ 0 0               RHBDD2 
##  6 1.57e-33      2.75 0.86  0.931  3.38e-29 0               UBL5   
##  7 6.54e- 9      2.34 0.794 0.855  1.41e- 4 0               ARL6IP1
##  8 1.64e- 5      2.14 0.633 0.679  3.53e- 1 0               DDX17  
##  9 7.89e-10      2.14 0.862 0.773  1.70e- 5 0               MAOB   
## 10 4.84e- 7      2.03 0.509 0.588  1.04e- 2 0               PSMC5  
## # ... with 6,431 more rows</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>
<code>stdf</code> refers to <em>summarized trajectory data.frames</em> which contain information about spatial trajectories drawn with <code><a href="../reference/createTrajectories.html">createTrajectories()</a></code> (obtained by e.g.<code><a href="../reference/getTrajectoryDf.html">getTrajectoryDf()</a></code>).</li>
</ol>
<pre><code>## # A tibble: 45 x 5
##    trajectory_part trajectory_part_order trajectory_order variables values
##    &lt;chr&gt;                           &lt;int&gt;            &lt;int&gt; &lt;chr&gt;      &lt;dbl&gt;
##  1 Part 1                              1                1 GFAP       1    
##  2 Part 1                              2                2 GFAP       0.972
##  3 Part 1                              3                3 GFAP       0.997
##  4 Part 1                              4                4 GFAP       0.932
##  5 Part 1                              5                5 GFAP       0.923
##  6 Part 1                              6                6 GFAP       0.904
##  7 Part 1                              7                7 GFAP       0.908
##  8 Part 1                              8                8 GFAP       0.968
##  9 Part 1                              9                9 GFAP       0.969
## 10 Part 1                             10               10 GFAP       0.824
## # ... with 35 more rows</code></pre>
<ol start="5" style="list-style-type: decimal">
<li>
<code>atdf</code> refers to <em>assessd trajectory data.frames</em> which contain the results spatial trajectory modelling. (obtained by e.g. <code><a href="../reference/assessTrajectoryTrends.html">assessTrajectoryTrends()</a></code>)</li>
</ol>
<pre><code>## # A tibble: 14 x 3
## # Groups:   variables [1]
##    variables  pattern                  auc
##    &lt;chr&gt;      &lt;chr&gt;                  &lt;dbl&gt;
##  1 HM_HYPOXIA Linear ascending        1.70
##  2 HM_HYPOXIA Gradient ascending      1.80
##  3 HM_HYPOXIA Sinus (reversed)        2.87
##  4 HM_HYPOXIA Late peak               3.45
##  5 HM_HYPOXIA One peak (reversed)     7.08
##  6 HM_HYPOXIA Logarithmic ascending   8.76
##  7 HM_HYPOXIA Two peaks               9.61
##  8 HM_HYPOXIA Logarithmic descending 10.5 
##  9 HM_HYPOXIA Two peaks (reversed)   10.7 
## 10 HM_HYPOXIA One peak               13.2 
## 11 HM_HYPOXIA Early peak             15.0 
## 12 HM_HYPOXIA Linear descending      15.1 
## 13 HM_HYPOXIA Sinus                  17.4 
## 14 HM_HYPOXIA Gradient descending    18.5</code></pre>
</div>
<div id="informative-variables" class="section level4">
<h4 class="hasAnchor">
<a href="#informative-variables" class="anchor"></a>4. Informative variables</h4>
<p>These are variables of the data.frame you specify as input of the <code>df</code>-argument family or of the data.frame that is generated from scratch in the background. Depending on what the function does they come along with different names but work in a similar fashion.</p>
<ol style="list-style-type: decimal">
<li><p><code>variables</code> refer to all types of informative variables as well and is used if more than one variable can be specified.</p></li>
<li><p><code>genes</code>/<code>gene_sets</code>/<code>features</code> take only input of the respective type.</p></li>
<li><p><code>discrete_feature</code>/<code>discrete_features</code> take only feature names as input whereby it must be a discrete/categorical one such as cluster belonging.</p></li>
</ol>
<p>(See <a href="spata-v2-genes-gene-sets-and-features.html">Genes-, Gene-sets &amp; Features</a> for more information.)</p>
</div>
<div id="comparing-arguments" class="section level4">
<h4 class="hasAnchor">
<a href="#comparing-arguments" class="anchor"></a>5. Comparing arguments</h4>
<ol style="list-style-type: decimal">
<li><p><code>across</code> takes one discrtete feature that denotes group belonging such as <em>clusters</em> or <em>segments</em> and is found in functions that compare certain aspects across these groups.</p></li>
<li><p><code>across_subset</code> takes specific values of the specified categorical feature variable in <code>across</code>.</p></li>
</ol>
<p>E.g. <code><a href="../reference/plotBoxplot.html">plotViolinplot()</a></code> with the following argument input <code>variables</code> = <code>c(*'GFAP'*, *'GAPDH'*)</code> <code>across</code> = <em>seurat_clusters</em> and <code>across_subset</code> = <code><a href="https://rdrr.io/r/base/c.html">c('0', '1', '2')</a></code> would compare the expression of the genes <em>GFAP</em> and <em>GAPDH</em> across the clusters named <em>0</em>, <em>1</em>, <em>2</em> of the clustering results that are stored in your feature data under the variable <em>seurat_clusters</em>.</p>
</div>
<div id="plotting" class="section level4">
<h4 class="hasAnchor">
<a href="#plotting" class="anchor"></a>6. Plotting</h4>
<p>Behind the scenes surface plots are scatterplots where the barcode-spots x- and y-coordinates are mapped onto the respective x- and y-aesthetic of the plot. These dots representing the barcode spots can be visually customized with the <code>pt_*</code>-arguments.</p>
<p>E.g. <code><a href="../reference/plotSurface.html">plotSurface()</a></code>with arguments <code>pt_size</code> = <em>2</em> and <code>pt_clrsp</code> = <em>‘inferno’</em> would generate a surfaceplot with dots of size 2 and with the colorspectrum <em>‘inferno’</em> used to display a continuous informative variable specified by <code>color_by</code> such as <em>‘GFAP’</em> expression levels.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Jan Kueckelhaus, Dieter-Henrik Heiland.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
