<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Genes-, Gene-sets &amp; Features â€¢ SPATA2</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Genes-, Gene-sets &amp; Features">
<meta property="og:description" content="SPATA2">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SPATA2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/spata-v2-introduction-and-philosophy.html">About</a>
</li>
<li>
  <a href="../articles/spata-v2-installation.html">Installation</a>
</li>
<li>
  <a href="../articles/spata-v2-terminology.html">Terminology</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Around the Spata-Object</li>
    <li>
      <a href="../articles/spata-v2-object-initiation-and-manipulation.html">Object Set Up and Manipulation</a>
    </li>
    <li>
      <a href="../articles/spata-v2-extract-add-and-join-data.html">Extracting, Adding and Joining Data</a>
    </li>
    <li>
      <a href="../articles/">Genes, Gene Sets and Features</a>
    </li>
    <li>
      <a href="../articles/spata-v2-platform-compatibility.html">Integrating Different Platforms</a>
    </li>
    <li class="dropdown-header">Analysis &amp; Computation</li>
    <li>
      <a href="../articles/spata-v2-autoencoder.html">Autoencoder Denoising</a>
    </li>
    <li>
      <a href="../articles/spata-v2-cnv-analysis.html">Copy Number Variations</a>
    </li>
    <li>
      <a href="../articles/spata-v2-segmentation-and-clustering.html">Spatial Segmentation &amp; Clustering</a>
    </li>
    <li>
      <a href="../articles/spata-v2-de-analysis.html">Differential Expression Analysis</a>
    </li>
    <li>
      <a href="../articles/spata-v2-gsea.html">Gene Set Enrichment Analysis</a>
    </li>
    <li>
      <a href="../articles/spata-v2-trajectory-analysis.html">Spatial Trajectories</a>
    </li>
    <li class="dropdown-header">Visualization</li>
    <li>
      <a href="../articles/spata-v2-plotting-surface.html">Surface Plotting</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-dimensional-reduction.html">Dimensional Reduction</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-descriptive-statistics.html">Descriptive Statistics</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-miscellaneous.html">Miscellaneous</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="spata-v2-genes-gene-sets-and-features_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Genes-, Gene-sets &amp; Features</h1>
            
      
      
      <div class="hidden name"><code>spata-v2-genes-gene-sets-and-features.Rmd</code></div>

    </div>

    
    
<p>Genes and gene-sets are of major importance when it comes to the analysis of every RNA sequencing results. The average count-matrix contains information of these expression levels for more than 20.000 genes. While referring to a few of them by entering their names manually might be tolerable things can get cumbersome rapidly once the number of genes of interest becomes double- or tripple digit. Still, the majority of functions in <strong>SPATA2</strong> needs information about the genes, gene-sets and features of interest in form of character vectors. The following tutorial introduces important helper functions that make living easier when it comes to working in <strong>SPATA2</strong> and working with these informative variables in general.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://themilolab.com/" class="external-link">SPATA2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span>

<span class="co"># load object</span>
<span class="va">spata_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/loadSpataObject.html">loadSpataObject</a></span><span class="op">(</span><span class="st">"data/spata-obj-gbm275.RDS"</span><span class="op">)</span></code></pre></div>
<div class="section level1">
<h1 id="genes-and-gene-sets">1. Genes and gene-sets<a class="anchor" aria-label="anchor" href="#genes-and-gene-sets"></a>
</h1>
<div class="section level2">
<h2 id="setting-up-a-gene-set-data-frame">1.1 Setting up a gene-set data.frame<a class="anchor" aria-label="anchor" href="#setting-up-a-gene-set-data-frame"></a>
</h2>
<p>The term gene-set refers to a collection of several genes that are known to interact with each other or being co-expressed together under certain conditions. Information about the gene-sets defined in your spata-object is stored in the slot <code>@used_genesets</code> which contains a data.frame of two character-variables.</p>
<ol style="list-style-type: decimal">
<li>
<em>ont</em>: The name of the gene-sets.</li>
<li>
<em>gene</em>: The belonging genes.</li>
</ol>
<pre><code><span class="co">##                ont     gene</span>
<span class="co">## 1  BC_RELA_PATHWAY    IKBKG</span>
<span class="co">## 2  BC_RELA_PATHWAY   NFKBIA</span>
<span class="co">## 3  BC_RELA_PATHWAY     FADD</span>
<span class="co">## 4  BC_RELA_PATHWAY     CHUK</span>
<span class="co">## 5  BC_RELA_PATHWAY      TNF</span>
<span class="co">## 6  BC_RELA_PATHWAY TNFRSF1B</span>
<span class="co">## 7  BC_RELA_PATHWAY TNFRSF1A</span>
<span class="co">## 8  BC_RELA_PATHWAY    TRAF6</span>
<span class="co">## 9  BC_RELA_PATHWAY   CREBBP</span>
<span class="co">## 10 BC_RELA_PATHWAY    TRADD</span>
<span class="co">## 11 BC_RELA_PATHWAY     RELA</span>
<span class="co">## 12 BC_RELA_PATHWAY    EP300</span>
<span class="co">## 13 BC_RELA_PATHWAY    RIPK1</span>
<span class="co">## 14 BC_RELA_PATHWAY    IKBKB</span>
<span class="co">## 15 BC_RELA_PATHWAY    NFKB1</span>
<span class="co">## 16  BC_NO1_PATHWAY    ACTA1</span>
<span class="co">## 17  BC_NO1_PATHWAY   SLC7A1</span>
<span class="co">## 18  BC_NO1_PATHWAY    VEGFA</span>
<span class="co">## 19  BC_NO1_PATHWAY     CAV1</span>
<span class="co">## 20  BC_NO1_PATHWAY     KNG1</span></code></pre>
<p>In <strong>SPATA2</strong> gene-set names are functionally divided into their class and their actual name. The class is determined by the character string that comes before the first *_*, the name refers to the rest. Subsequent chapters will highlight why this functional separation is useful as well as why it is useful to really think about the way you name your personally defined gene-sets. In order to obtain an overview of your saved gene-sets run <code><a href="../reference/printGeneSetOverview.html">printGeneSetOverview()</a></code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/printGeneSetOverview.html">printGeneSetOverview</a></span><span class="op">(</span><span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##    Class Available Gene Sets</span>
<span class="co">## 1     BC                 289</span>
<span class="co">## 2  BP.GO                7269</span>
<span class="co">## 3  CC.GO                 972</span>
<span class="co">## 4     HM                  50</span>
<span class="co">## 5  MF.GO                1581</span>
<span class="co">## 6 Neftel                  10</span>
<span class="co">## 7   RCTM                1493</span></code></pre>
<p>By default any <code>initiateSpataObject_*()</code>-function assigns a gene-set data.frame to your spata-object containing a variety of predefined gene-sets from the <a href="https://www.gsea-msigdb.org/gsea/index.jsp" class="external-link">Molecular Signature Database</a>. Run <code>data(gsdf)</code> to load and <code><a href="../reference/gsdf.html">?gsdf</a></code> to get information about it and what the class-names stand for. If you want to add new ones you can do that via <code><a href="../reference/addGeneSet.html">addGeneSet()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># add a new gene-set of three genes</span>
<span class="va">spata_obj</span> <span class="op">&lt;-</span> 
  <span class="fu"><a href="../reference/addGeneSet.html">addGeneSet</a></span><span class="op">(</span>
    object <span class="op">=</span> <span class="va">spata_obj</span>,
    class_name <span class="op">=</span> <span class="st">"New"</span>, 
    gs_name <span class="op">=</span> <span class="st">"example_1"</span>, 
    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'SLC35E2A'</span>, <span class="st">'NADK'</span>, <span class="st">'GFAP'</span><span class="op">)</span>
  <span class="op">)</span> <span class="co"># genes that exist in your expression matrix</span>

<span class="fu"><a href="../reference/printGeneSetOverview.html">printGeneSetOverview</a></span><span class="op">(</span><span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##    Class Available Gene Sets</span>
<span class="co">## 1     BC                 289</span>
<span class="co">## 2  BP.GO                7269</span>
<span class="co">## 3  CC.GO                 972</span>
<span class="co">## 4     HM                  50</span>
<span class="co">## 5  MF.GO                1581</span>
<span class="co">## 6 Neftel                  10</span>
<span class="co">## 7    New                   1</span>
<span class="co">## 8   RCTM                1493</span></code></pre>
<p>A more convenient way would be to open an interactive application via <code>addGeneSetsInteracive()</code> as displayed below. Make sure to close the application via the respective button and to store the resulting object in the same object in order to overwrite it and not to overcrowd your global environment with spata-objects that only differ in a few gene-sets.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">spata_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addGeneSet.html">addGeneSetsInteractive</a></span><span class="op">(</span><span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<div class="figure" style="text-align: left">
<img src="images/genes-and-gene-sets-fig-1.png" alt="Figure 1.1 Interface of addGeneSetsInteractive()" width="100%"><p class="caption">
Figure 1.1 Interface of addGeneSetsInteractive()
</p>
</div>
<p>The gene-set data.frame is infinitely expandable. If you want to discard gene-sets or alter them you can either use <code><a href="../reference/addGeneSet.html">addGeneSet()</a></code> with argument <code>overwrite</code> set to TRUE in order to overwrite existing ones or <code><a href="../reference/discardGeneSets.html">discardGeneSets()</a></code> as shown below.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># discard gene-sets by name </span>
<span class="va">spata_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/discardGeneSets.html">discardGeneSets</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, gs_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"New_example_2"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># display updated overview</span>
<span class="fu"><a href="../reference/printGeneSetOverview.html">printGeneSetOverview</a></span><span class="op">(</span><span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##    Class Available Gene Sets</span>
<span class="co">## 1     BC                 289</span>
<span class="co">## 2  BP.GO                7269</span>
<span class="co">## 3  CC.GO                 972</span>
<span class="co">## 4     HM                  50</span>
<span class="co">## 5  MF.GO                1581</span>
<span class="co">## 6 Neftel                  10</span>
<span class="co">## 7    New                   1</span>
<span class="co">## 8   RCTM                1493</span></code></pre>
<p>In order to save the gene-set data.frame of one particular spata-object for further use with other spata-objects use <code><a href="../reference/saveGeneSetDf.html">saveGeneSetDf()</a></code> which saves the <code>@used_genesets</code> slot of the specified object as a .RDS-file.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/saveGeneSetDf.html">saveGeneSetDf</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, directory <span class="op">=</span> <span class="st">"data/gene-set-df.RDS"</span><span class="op">)</span> </code></pre></div>
<p>If you want to set your gene-set data.frame manually use <code><a href="../reference/setGeneSetDf.html">setGeneSetDf()</a></code>.</p>
</div>
<div class="section level2">
<h2 id="adjusting-a-gene-set-data-frame">1.2 Adjusting a gene-set data.frame<a class="anchor" aria-label="anchor" href="#adjusting-a-gene-set-data-frame"></a>
</h2>
<p>Given the nature of your sample it will happen that your expression matrix does not contain genes (0 reads across all barcode-spots) although they appear in certain gene-sets. <code><a href="../reference/joinWith.html">joinWithGeneSets()</a></code> which works under the hood of a variety of functions does not allow for gene-sets to join your data if less than 25% of the genes needed to form that gene-set are found. If this threshold is to loose to you you can use <code><a href="../reference/adjustGeneSetDf.html">adjustGeneSetDf()</a></code> to set the limit yourself. It will then discard all gene-sets whoose belonging genes are found to a percentage lower than the threshold you set. (Discarding gene-sets from your gene-set data.frame effectively makes it impossible to work with them within <strong>SPATA2</strong>.)</p>
</div>
<div class="section level2">
<h2 id="referring-to-genes-and-gene-sets">1.3 Referring to genes and gene-sets<a class="anchor" aria-label="anchor" href="#referring-to-genes-and-gene-sets"></a>
</h2>
<p><code><a href="../reference/getGenes.html">getGenes()</a></code> and <code><a href="../reference/getGeneSets.html">getGeneSets()</a></code> return character vectors which are valid inputs for all <strong>SPATA2</strong>-functions in which one has to refer to genes or gene-sets in any way.</p>
<div class="section level3">
<h3 id="obtaining-gene-set-names">1.3.1 Obtaining gene-set names<a class="anchor" aria-label="anchor" href="#obtaining-gene-set-names"></a>
</h3>
<p>As we have seen above gene-sets are a set of genes that are functionally gathered under a certain name. <code><a href="../reference/getGeneSets.html">getGeneSets()</a></code> <strong>does not</strong> return the genes of certain gene-sets <strong>but</strong> the gene-set-names as strings It does that either as a single character vector or as a named list.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># to get all gene-set names set 'of_class' to 'all'</span>
<span class="va">all_gs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getGeneSets.html">getGeneSets</a></span><span class="op">(</span><span class="va">spata_obj</span>, of_class <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">all_gs</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 11665</span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">all_gs</span>, <span class="fl">10</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  [1] "BC_RELA_PATHWAY"         "BC_NO1_PATHWAY"         </span>
<span class="co">##  [3] "BC_CSK_PATHWAY"          "BC_SRCTMRPTP_PATHWAY"   </span>
<span class="co">##  [5] "BC_AMI_PATHWAY"          "BC_GRANULOCYTES_PATHWAY"</span>
<span class="co">##  [7] "BC_LYM_PATHWAY"          "BC_ARAP_PATHWAY"        </span>
<span class="co">##  [9] "BC_AGR_PATHWAY"          "BC_AKAP95_PATHWAY"</span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get gene-set names of certain classes </span>
<span class="fu"><a href="../reference/getGeneSets.html">getGeneSets</a></span><span class="op">(</span>
  object <span class="op">=</span> <span class="va">spata_obj</span>,
  of_class <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HM"</span>, <span class="st">"New"</span><span class="op">)</span>,
  simplify <span class="op">=</span> <span class="cn">TRUE</span> <span class="co"># as a single vector</span>
  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># show last six </span></code></pre></div>
<pre><code><span class="co">## [1] "HM_ALLOGRAFT_REJECTION" "HM_SPERMATOGENESIS"     "HM_KRAS_SIGNALING_UP"  </span>
<span class="co">## [4] "HM_KRAS_SIGNALING_DN"   "HM_PANCREAS_BETA_CELLS" "New_example_1"</span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get gene-set names of certain classes </span>
<span class="fu"><a href="../reference/getGeneSets.html">getGeneSets</a></span><span class="op">(</span><span class="va">spata_obj</span>,
            of_class <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HM"</span>, <span class="st">"New"</span><span class="op">)</span>,
            simplify <span class="op">=</span> <span class="cn">FALSE</span> <span class="co"># as a named list</span>
            <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">head</span><span class="op">)</span> <span class="co">#show first six</span></code></pre></div>
<pre><code><span class="co">## $HM</span>
<span class="co">## [1] "HM_TNFA_SIGNALING_VIA_NFKB"    "HM_HYPOXIA"                   </span>
<span class="co">## [3] "HM_CHOLESTEROL_HOMEOSTASIS"    "HM_MITOTIC_SPINDLE"           </span>
<span class="co">## [5] "HM_WNT_BETA_CATENIN_SIGNALING" "HM_TGF_BETA_SIGNALING"        </span>
<span class="co">## </span>
<span class="co">## $New</span>
<span class="co">## [1] "New_example_1"</span></code></pre>
<p>To subset your gene-sets of interest additionally make use of the argument <code>index</code> which takes a character string as a regular expression and filters the gene-sets to be returned again according to it.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get all gene-sets of class HALLMARK referring to something specific</span>
<span class="fu"><a href="../reference/getGeneSets.html">getGeneSets</a></span><span class="op">(</span><span class="va">spata_obj</span>,
            of_class <span class="op">=</span> <span class="st">"HM"</span>,
            index <span class="op">=</span> <span class="st">"ESTROGEN|ANDROGEN"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "HM_ESTROGEN_RESPONSE_EARLY" "HM_ESTROGEN_RESPONSE_LATE" </span>
<span class="co">## [3] "HM_ANDROGEN_RESPONSE"</span></code></pre>
<p>If you want to skim your gene-sets interactively use <code><a href="../reference/getGeneSets.html">getGeneSetsInteractive()</a></code> which opens an application in your R-Studioâ€™s viewer pane and returns a vector of chosen gene-sets.</p>
</div>
<div class="section level3">
<h3 id="obtaining-gene-names">1.3.2 Obtaining gene names<a class="anchor" aria-label="anchor" href="#obtaining-gene-names"></a>
</h3>
<p>Eventually expression matrices contain information about gene expression levels and not gene-set expression levels. If you are looking for genes of certain gene-sets <code><a href="../reference/getGenes.html">getGenes()</a></code> provides an easy we to do just that.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># getGenes() without any further specification will return all genes found</span>
<span class="co"># in your expression matrix ...</span>
<span class="va">all_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getGenes.html">getGenes</a></span><span class="op">(</span><span class="va">spata_obj</span><span class="op">)</span>

<span class="co"># ... which can be quite a lot</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">all_genes</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 21548</span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># with regards to gene-sets it makes more sense to break genes down by </span>
<span class="co"># their gene-set belonging</span>
<span class="va">estrogen_gs</span> <span class="op">&lt;-</span> 
  <span class="fu"><a href="../reference/getGeneSets.html">getGeneSets</a></span><span class="op">(</span><span class="va">spata_obj</span>,
              of_class <span class="op">=</span> <span class="st">"HM"</span>,
              index <span class="op">=</span> <span class="st">"ESTROGEN"</span><span class="op">)</span>

<span class="va">estrogen_genes</span> <span class="op">&lt;-</span> 
  <span class="fu"><a href="../reference/getGenes.html">getGenes</a></span><span class="op">(</span><span class="va">spata_obj</span>, 
           of_gene_sets <span class="op">=</span> <span class="va">estrogen_gs</span>,
           simplify <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="co"># to return them sorted in a list</span>

<span class="co">#hint: "..." was added manually for visualization purpose as these gene-sets contain </span>
<span class="co"># more than 150 genes </span></code></pre></div>
<pre><code><span class="co">## $HM_ESTROGEN_RESPONSE_EARLY</span>
<span class="co">##  [1] "GREB1"    "CA12"     "SLC9A3R1" "MYB"      "ANXA9"    "IGFBP4"  </span>
<span class="co">##  [7] "SYBU"     "NPY1R"    "PDZK1"    "NRIP1"    "MLPH"     "HSPB8"   </span>
<span class="co">## [13] "EGR3"     "KRT19"    "LRIG1"    "KDM4B"    "PGR"      "RHOBTB3" </span>
<span class="co">## [19] "TPD52L1"  "ELOVL2"   "..."     </span>
<span class="co">## </span>
<span class="co">## $HM_ESTROGEN_RESPONSE_LATE</span>
<span class="co">##  [1] "SLC9A3R1" "TPD52L1"  "PRSS23"   "CA12"     "PDZK1"    "ANXA9"   </span>
<span class="co">##  [7] "CELSR2"   "PGR"      "RET"      "MYB"      "TPBG"     "EGR3"    </span>
<span class="co">## [13] "ARL3"     "OLFM1"    "NPY1R"    "SCNN1A"   "XBP1"     "AREG"    </span>
<span class="co">## [19] "IL17RB"   "NRIP1"    "..."</span></code></pre>
<p>If you want to work with genes manually and skim them interactively use <code><a href="../reference/getGenes.html">getGenesInteractive()</a></code> which opens an application in your R-Studioâ€™s viewer pane.</p>
</div>
</div>
</div>
<div class="section level1">
<h1 id="features">2. Features<a class="anchor" aria-label="anchor" href="#features"></a>
</h1>
<p>In <strong>SPATA2</strong> features refer to all kinds of information of barcode-spots that do not fall explicitly under gene- or gene-set expression levels. This includes in particular group belonging such as clustering or spatial segmentation. But additional numeric features such as monocle3-pseudotime or everything else that has been computed is considered to be a feature as well. Feature information is stored - in a tidy data fashion - in the slot <code>@fdata</code>.</p>
<div class="section level2">
<h2 id="obtain-feature-data">2.1 Obtain feature data<a class="anchor" aria-label="anchor" href="#obtain-feature-data"></a>
</h2>
<p>To obtain all feature information use <code><a href="../reference/getFeatureDf.html">getFeatureDf()</a></code>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getFeatureDf.html">getFeatureDf</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="color: #949494;"># A tibble: 3,734 x 9</span></span>
<span class="co">##    barcodes         sample nCount_Spatial nFeature_Spatial percent.mt percent.RB</span>
<span class="co">##    <span style="color: #949494;font-style: italic;">&lt;chr&gt;</span><span>            </span><span style="color: #949494;font-style: italic;">&lt;chr&gt;</span><span>           </span><span style="color: #949494;font-style: italic;">&lt;dbl&gt;</span><span>            </span><span style="color: #949494;font-style: italic;">&lt;int&gt;</span><span>      </span><span style="color: #949494;font-style: italic;">&lt;dbl&gt;</span><span>      </span><span style="color: #949494;font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 1</span><span> AAACAAGTATCTCCC~ T275             </span><span style="text-decoration: underline;">4</span><span>572             </span><span style="text-decoration: underline;">2</span><span>173      11.6        5.60</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 2</span><span> AAACACCAATAACTG~ T275             </span><span style="text-decoration: underline;">6</span><span>303             </span><span style="text-decoration: underline;">2</span><span>856      11.3        4.82</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 3</span><span> AAACAGAGCGACTCC~ T275             </span><span style="text-decoration: underline;">5</span><span>999             </span><span style="text-decoration: underline;">2</span><span>840       7.53       3.50</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 4</span><span> AAACAGCTTTCAGAA~ T275             </span><span style="text-decoration: underline;">8</span><span>736             </span><span style="text-decoration: underline;">3</span><span>710       4.66       3.72</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 5</span><span> AAACAGGGTCTATAT~ T275             </span><span style="text-decoration: underline;">8</span><span>512             </span><span style="text-decoration: underline;">3</span><span>763       6.13       3.87</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 6</span><span> AAACAGTGTTCCTGG~ T275             </span><span style="text-decoration: underline;">7</span><span>970             </span><span style="text-decoration: underline;">3</span><span>305      10.8        4.89</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 7</span><span> AAACATTTCCCGGAT~ T275             </span><span style="text-decoration: underline;">4</span><span>495             </span><span style="text-decoration: underline;">2</span><span>238      11.0        3.98</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 8</span><span> AAACCGGGTAGGTAC~ T275             </span><span style="text-decoration: underline;">8</span><span>251             </span><span style="text-decoration: underline;">3</span><span>438       5.55       4.65</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 9</span><span> AAACCGTTCGTCCAG~ T275             </span><span style="text-decoration: underline;">4</span><span>688             </span><span style="text-decoration: underline;">2</span><span>318       8.79       4.69</span></span>
<span class="co">## <span style="color: #BCBCBC;">10</span><span> AAACCTAAGCAGCCG~ T275             </span><span style="text-decoration: underline;">9</span><span>070             </span><span style="text-decoration: underline;">3</span><span>909       9.18       3.78</span></span>
<span class="co">## <span style="color: #949494;"># ... with 3,724 more rows, and 3 more variables: seurat_clusters &lt;fct&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   segmentation &lt;fct&gt;, igraph_clusters &lt;chr&gt;</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="obtain-feature-names-variables-and-values">2.2 Obtain feature names, variables and values<a class="anchor" aria-label="anchor" href="#obtain-feature-names-variables-and-values"></a>
</h2>
<p>In order to specify features of interest as input in functions you need to specify the name that particular feature variable carries. Several functions only work with feature data of one kind - either numeric features or discrete/categorical features. The function <code><a href="../reference/getFeatureNames.html">getFeatureNames()</a></code> provides you with information about all currently available features in your spata-object as well as about the class they belong to.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get all feature names</span>
<span class="fu"><a href="../reference/getFeatureNames.html">getFeatureNames</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##            numeric            integer            numeric            numeric </span>
<span class="co">##   "nCount_Spatial" "nFeature_Spatial"       "percent.mt"       "percent.RB" </span>
<span class="co">##             factor             factor          character </span>
<span class="co">##  "seurat_clusters"     "segmentation"  "igraph_clusters"</span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get only names of numeric features</span>
<span class="va">numeric_features</span> <span class="op">&lt;-</span> 
  <span class="fu"><a href="../reference/getFeatureNames.html">getFeatureNames</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, of_class <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"numeric"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># output</span>
<span class="va">numeric_features</span></code></pre></div>
<pre><code><span class="co">##          numeric          numeric          numeric </span>
<span class="co">## "nCount_Spatial"     "percent.mt"     "percent.RB"</span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get only names of discrete features</span>
<span class="va">discrete_features</span> <span class="op">&lt;-</span> 
  <span class="fu"><a href="../reference/getFeatureNames.html">getFeatureNames</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, of_class <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"character"</span>, <span class="st">"factor"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># output</span>
<span class="va">discrete_features</span></code></pre></div>
<pre><code><span class="co">##            factor            factor         character </span>
<span class="co">## "seurat_clusters"    "segmentation" "igraph_clusters"</span></code></pre>
<p>Additionally <code><a href="../reference/getFeatureValues.html">getFeatureValues()</a></code> allows to quickly access the unique values of discrete feature variables such as clusters.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getFeatureValues.html">getFeatureValues</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, features <span class="op">=</span> <span class="va">discrete_features</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## $seurat_clusters</span>
<span class="co">## [1] 0 2 1 3 5 6 4 7 8</span>
<span class="co">## Levels: 0 1 2 3 4 5 6 7 8</span>
<span class="co">## </span>
<span class="co">## $segmentation</span>
<span class="co">## [1] none             oxid-phosph-high</span>
<span class="co">## Levels: none oxid-phosph-high</span>
<span class="co">## </span>
<span class="co">## $igraph_clusters</span>
<span class="co">## [1] "Cluster 1" "Cluster 3" "Cluster 4" "Cluster 7" "Cluster 6" "Cluster 2"</span>
<span class="co">## [7] "Cluster 5"</span></code></pre>
<p>When it comes to grouping variables such as clusters a more intuitive approach is offered by the two functions <code><a href="../reference/getGroupingOptions.html">getGroupingOptions()</a></code> and <code><a href="../reference/getGroupNames.html">getGroupNames()</a></code>.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># obtain all variables that group the sample's barcode-spots in a certain way</span>
<span class="fu"><a href="../reference/getGroupingOptions.html">getGroupingOptions</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##            factor            factor         character </span>
<span class="co">## "seurat_clusters"    "segmentation" "igraph_clusters"</span></code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># obtain the names of the groups a certain grouping variable contains </span>
<span class="fu"><a href="../reference/getGroupNames.html">getGroupNames</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, discrete_feature <span class="op">=</span> <span class="st">"seurat_clusters"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "0" "1" "2" "3" "4" "5" "6" "7" "8"</span></code></pre>
</div>
<div class="section level2">
<h2 id="alter-features">2.3 Alter features<a class="anchor" aria-label="anchor" href="#alter-features"></a>
</h2>
<div class="section level3">
<h3 id="change-feature-names">2.3.1 Change feature names<a class="anchor" aria-label="anchor" href="#change-feature-names"></a>
</h3>
<p>If you want to change the name of a feature variable you can do so with <code><a href="../reference/renameFeatures.html">renameFeatures()</a></code>.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># old feature names</span>
<span class="fu"><a href="../reference/getFeatureNames.html">getFeatureNames</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##            numeric            integer            numeric            numeric </span>
<span class="co">##   "nCount_Spatial" "nFeature_Spatial"       "percent.mt"       "percent.RB" </span>
<span class="co">##             factor             factor          character </span>
<span class="co">##  "seurat_clusters"     "segmentation"  "igraph_clusters"</span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># change some names</span>
<span class="va">spata_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/renameFeatures.html">renameFeatures</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
                            <span class="st">"seurat_cl"</span> <span class="op">=</span> <span class="st">"seurat_clusters"</span>, 
                            <span class="st">"nCount"</span> <span class="op">=</span> <span class="st">"nCount_Spatial"</span>
                            <span class="op">)</span>

<span class="co"># new feature names</span>
<span class="fu"><a href="../reference/getFeatureNames.html">getFeatureNames</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##            numeric            integer            numeric            numeric </span>
<span class="co">##           "nCount" "nFeature_Spatial"       "percent.mt"       "percent.RB" </span>
<span class="co">##             factor             factor          character </span>
<span class="co">##        "seurat_cl"     "segmentation"  "igraph_clusters"</span></code></pre>
</div>
<div class="section level3">
<h3 id="change-feature-values-group-names">2.3.2 Change feature values (group names)<a class="anchor" aria-label="anchor" href="#change-feature-values-group-names"></a>
</h3>
<p>Referring to clusters/groups with numbers is often suboptimal. Once your differential gene expression analysis has given you insight in what constitutes certain groups of barcode-spots you might want to give it a more informative name. You can do so with <code><a href="../reference/renameGroups.html">renameGroups()</a></code>.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># old group names</span>
<span class="fu"><a href="../reference/getGroupNames.html">getGroupNames</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, discrete_feature <span class="op">=</span> <span class="st">"seurat_cl"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "0" "1" "2" "3" "4" "5" "6" "7" "8"</span></code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># rename cluster </span>
<span class="va">spata_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/renameGroups.html">renameGroups</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>,
                          discrete_feature <span class="op">=</span> <span class="st">"seurat_cl"</span>,
                          <span class="st">"reactive-inflammatory"</span> <span class="op">=</span> <span class="st">"3"</span>,
                          <span class="st">"reactive-hypoxia"</span> <span class="op">=</span> <span class="st">"4"</span>
                          <span class="op">)</span>

<span class="fu"><a href="../reference/getGroupNames.html">getGroupNames</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, discrete_feature <span class="op">=</span> <span class="st">"seurat_cl"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "0"                     "1"                     "2"                    </span>
<span class="co">## [4] "reactive-inflammatory" "reactive-hypoxia"      "5"                    </span>
<span class="co">## [7] "6"                     "7"                     "8"</span></code></pre>
<p>Whenever a warning message appears telling you that a certain gene, gene-set or feature was not found this means that the character string provided was not found throughout your spata-object. Check for typos in this case. The same is true for gene-set names with regards to the <em>ont</em>-variable in the gene-set data.frame.</p>
</div>
</div>
</div>
<div class="section level1">
<h1 id="example">3. Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h1>
<p>The output vectors of <code><a href="../reference/getGenes.html">getGenes()</a></code>, <code><a href="../reference/getGeneSets.html">getGeneSets()</a></code> and <code><a href="../reference/getFeatureNames.html">getFeatureNames()</a></code> are a perfectly valid inputs for arguments like <code>color_by</code> or <code>variables</code>.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hallmark_gs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getGeneSets.html">getGeneSets</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, of_class <span class="op">=</span> <span class="st">"HM"</span><span class="op">)</span>

<span class="co"># output</span>
<span class="va">hallmark_gs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></code></pre></div>
<pre><code><span class="co">##  [1] "HM_TNFA_SIGNALING_VIA_NFKB"    "HM_HYPOXIA"                   </span>
<span class="co">##  [3] "HM_CHOLESTEROL_HOMEOSTASIS"    "HM_MITOTIC_SPINDLE"           </span>
<span class="co">##  [5] "HM_WNT_BETA_CATENIN_SIGNALING" "HM_TGF_BETA_SIGNALING"        </span>
<span class="co">##  [7] "HM_IL6_JAK_STAT3_SIGNALING"    "HM_DNA_REPAIR"                </span>
<span class="co">##  [9] "HM_G2M_CHECKPOINT"             "HM_APOPTOSIS"</span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># create a spata-data.frame</span>
<span class="va">joined_df</span> <span class="op">&lt;-</span> 
  <span class="fu"><a href="../reference/joinWith.html">joinWith</a></span><span class="op">(</span>
    object <span class="op">=</span> <span class="va">spata_obj</span>, 
    spata_df <span class="op">=</span> <span class="fu"><a href="../reference/getCoordsDf.html">getCoordsDf</a></span><span class="op">(</span><span class="va">spata_obj</span><span class="op">)</span>,
    gene_sets <span class="op">=</span> <span class="va">hallmark_gs</span>,
    features <span class="op">=</span> <span class="va">discrete_features</span>
  <span class="op">)</span>

<span class="co">#output</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">joined_df</span>, <span class="fl">20</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="color: #949494;"># A tibble: 20 x 56</span></span>
<span class="co">##    barcodes   sample     x     y segmentation  igraph_clusters HM_TNFA_SIGNALIN~</span>
<span class="co">##    <span style="color: #949494;font-style: italic;">&lt;chr&gt;</span><span>      </span><span style="color: #949494;font-style: italic;">&lt;chr&gt;</span><span>  </span><span style="color: #949494;font-style: italic;">&lt;dbl&gt;</span><span> </span><span style="color: #949494;font-style: italic;">&lt;dbl&gt;</span><span> </span><span style="color: #949494;font-style: italic;">&lt;fct&gt;</span><span>         </span><span style="color: #949494;font-style: italic;">&lt;chr&gt;</span><span>                       </span><span style="color: #949494;font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 1</span><span> AAACAAGTA~ T275   435.  374.  none          Cluster 1                   0.289</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 2</span><span> AAACACCAA~ T275   132.  429.  oxid-phosph-~ Cluster 3                   0.243</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 3</span><span> AAACAGAGC~ T275   407.  145.  none          Cluster 3                   0.384</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 4</span><span> AAACAGCTT~ T275    96   327   none          Cluster 1                   0.583</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 5</span><span> AAACAGGGT~ T275   110.  353.  none          Cluster 3                   0.518</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 6</span><span> AAACAGTGT~ T275   219.  518.  oxid-phosph-~ Cluster 4                   0.282</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 7</span><span> AAACATTTC~ T275   416.  443.  none          Cluster 3                   0.344</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 8</span><span> AAACCGGGT~ T275   165.  321.  none          Cluster 7                   0.540</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 9</span><span> AAACCGTTC~ T275   216.  385.  oxid-phosph-~ Cluster 7                   0.316</span></span>
<span class="co">## <span style="color: #BCBCBC;">10</span><span> AAACCTAAG~ T275   365.  468.  none          Cluster 6                   0.346</span></span>
<span class="co">## <span style="color: #BCBCBC;">11</span><span> AAACCTCAT~ T275   133.  289.  none          Cluster 4                   0.595</span></span>
<span class="co">## <span style="color: #BCBCBC;">12</span><span> AAACGAAGA~ T275   298.   93.7 none          Cluster 7                   0.263</span></span>
<span class="co">## <span style="color: #BCBCBC;">13</span><span> AAACGAAGA~ T275    77.2 422.  oxid-phosph-~ Cluster 4                   0.328</span></span>
<span class="co">## <span style="color: #BCBCBC;">14</span><span> AAACGAGAC~ T275   352.  278.  none          Cluster 1                   0.482</span></span>
<span class="co">## <span style="color: #BCBCBC;">15</span><span> AAACGGGCG~ T275   394.  469.  none          Cluster 2                   0.400</span></span>
<span class="co">## <span style="color: #BCBCBC;">16</span><span> AAACGGTTG~ T275   277.  480.  none          Cluster 3                   0.281</span></span>
<span class="co">## <span style="color: #BCBCBC;">17</span><span> AAACTCGGT~ T275   317.  474.  none          Cluster 6                   0.305</span></span>
<span class="co">## <span style="color: #BCBCBC;">18</span><span> AAACTGCTG~ T275   307.  341.  none          Cluster 6                   0.338</span></span>
<span class="co">## <span style="color: #BCBCBC;">19</span><span> AAACTTAAT~ T275   106.  460.  oxid-phosph-~ Cluster 4                   0.305</span></span>
<span class="co">## <span style="color: #BCBCBC;">20</span><span> AAACTTGCA~ T275   132.  340.  none          Cluster 1                   0.578</span></span>
<span class="co">## <span style="color: #949494;"># ... with 49 more variables: HM_HYPOXIA &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_CHOLESTEROL_HOMEOSTASIS &lt;dbl&gt;, HM_MITOTIC_SPINDLE &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_WNT_BETA_CATENIN_SIGNALING &lt;dbl&gt;, HM_TGF_BETA_SIGNALING &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_IL6_JAK_STAT3_SIGNALING &lt;dbl&gt;, HM_DNA_REPAIR &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_G2M_CHECKPOINT &lt;dbl&gt;, HM_APOPTOSIS &lt;dbl&gt;, HM_NOTCH_SIGNALING &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_ADIPOGENESIS &lt;dbl&gt;, HM_ESTROGEN_RESPONSE_EARLY &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_ESTROGEN_RESPONSE_LATE &lt;dbl&gt;, HM_ANDROGEN_RESPONSE &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_MYOGENESIS &lt;dbl&gt;, HM_PROTEIN_SECRETION &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_INTERFERON_ALPHA_RESPONSE &lt;dbl&gt;, HM_INTERFERON_GAMMA_RESPONSE &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_APICAL_JUNCTION &lt;dbl&gt;, HM_APICAL_SURFACE &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_HEDGEHOG_SIGNALING &lt;dbl&gt;, HM_COMPLEMENT &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_UNFOLDED_PROTEIN_RESPONSE &lt;dbl&gt;, HM_PI3K_AKT_MTOR_SIGNALING &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_MTORCTM1_SIGNALING &lt;dbl&gt;, HM_E2F_TARGETS &lt;dbl&gt;, HM_MYC_TARGETS_V1 &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_MYC_TARGETS_V2 &lt;dbl&gt;, HM_EPITHELIAL_MESENCHYMAL_TRANSITION &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_INFLAMMATORY_RESPONSE &lt;dbl&gt;, HM_XENOBIOTIC_METABOLISM &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_FATTY_ACID_METABOLISM &lt;dbl&gt;, HM_OXIDATIVE_PHOSPHORYLATION &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_GLYCOLYSIS &lt;dbl&gt;, HM_REACTIVE_OXYGEN_SPECIES_PATHWAY &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_P53_PATHWAY &lt;dbl&gt;, HM_UV_RESPONSE_UP &lt;dbl&gt;, HM_UV_RESPONSE_DN &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_ANGIOGENESIS &lt;dbl&gt;, HM_HEME_METABOLISM &lt;dbl&gt;, HM_COAGULATION &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_IL2_STAT5_SIGNALING &lt;dbl&gt;, HM_BILE_ACID_METABOLISM &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_PEROXISOME &lt;dbl&gt;, HM_ALLOGRAFT_REJECTION &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_SPERMATOGENESIS &lt;dbl&gt;, HM_KRAS_SIGNALING_UP &lt;dbl&gt;,</span></span>
<span class="co">## <span style="color: #949494;">#   HM_KRAS_SIGNALING_DN &lt;dbl&gt;, HM_PANCREAS_BETA_CELLS &lt;dbl&gt;</span></span></code></pre>
<p>With our vector <code>hallmark_gs</code> containing our gene-sets of interest we can work throughout our R-session.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotBoxplot.html">plotRidgeplot</a></span><span class="op">(</span>
  object <span class="op">=</span> <span class="va">spata_obj</span>,
  variables <span class="op">=</span> <span class="va">hallmark_gs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span>,
  across <span class="op">=</span> <span class="st">"seurat_cl"</span>,
  across_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"reactive-hypoxia"</span>, <span class="st">"reactive-inflammatory"</span><span class="op">)</span>,
  ncol <span class="op">=</span> <span class="fl">2</span>
  <span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-genes-gene-sets-and-features_files/figure-html/application-demo-2-1.png" alt="Figure 3.1. Example plot on how to use getGeneSets()" width="100%"><p class="caption">
Figure 3.1. Example plot on how to use getGeneSets()
</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jan Kueckelhaus, Dieter-Henrik Heiland.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
