<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>SPATA &amp; Other Platforms • SPATA2</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="SPATA &amp; Other Platforms">
<meta property="og:description" content="SPATA2">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SPATA2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/spata-v2-introduction-and-philosophy.html">About</a>
</li>
<li>
  <a href="../articles/spata-v2-installation.html">Installation</a>
</li>
<li>
  <a href="../articles/spata-v2-terminology.html">Terminology</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Around the Spata-Object</li>
    <li>
      <a href="../articles/spata-v2-object-initiation-and-manipulation.html">Object Set Up and Manipulation</a>
    </li>
    <li>
      <a href="../articles/spata-v2-extract-add-and-join-data.html">Extracting, Adding and Joining Data</a>
    </li>
    <li>
      <a href="../articles/spata-v2-platform-compatibility.html">Integrating Different Platforms</a>
    </li>
    <li class="dropdown-header">Analysis &amp; Computation</li>
    <li>
      <a href="../articles/spata-v2-autoencoder.html">Autoencoder Denoising</a>
    </li>
    <li>
      <a href="../articles/spata-v2-de-analysis.html">Differential Expression Analysis</a>
    </li>
    <li>
      <a href="../articles/spata-v2-segmentation-and-clustering.html">Spatial Segmentation &amp; Clustering</a>
    </li>
    <li>
      <a href="../articles/spata-v2-trajectory-analysis.html">Spatial Trajectories</a>
    </li>
    <li class="dropdown-header">Visualization</li>
    <li>
      <a href="../articles/spata-v2-plotting-surface.html">Surface Plotting</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-dimensional-reduction.html">Dimensional Reduction</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-descriptive-statistics.html">Descriptive Statistics</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-miscellaneous.html">Miscellaneous</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="spata-v2-platform-compatibility_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>SPATA &amp; Other Platforms</h1>
            
      
      
      <div class="hidden name"><code>spata-v2-platform-compatibility.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>1. Introduction</h2>
<p>Apart from <strong>SPATA2</strong> there have been published several packages that devote themself to the analysis of sequencing data. Two of them are <code>monocle3</code> and <code>seurat</code>. This article demonstrates how to leverage <strong>SPATA2</strong>’s compatibility.</p>
</div>
<div id="monocle3" class="section level2">
<h2 class="hasAnchor">
<a href="#monocle3" class="anchor"></a>2. Monocle3</h2>
<p>The package <a href="https://cole-trapnell-lab.github.io/monocle3/"><code>monocle3</code></a> provides a variety of tools to analyze and visualize single-cell-sequencing data. Though spatial transcriptomic experiments do not reach the resolution of single-cell-sequencing yet you can leverage a variety of monocle3’s functions for your analysis and easily integrate the results back to your spata-object.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://themilolab.com/">SPATA2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org">magrittr</a></span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com">patchwork</a></span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">monocle3</span><span class="op">)</span>

<span class="co"># load example of human brain cortex</span>
<span class="va">spata_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/loadSpataObject.html">loadSpataObject</a></span><span class="op">(</span><span class="st">"data/spata-obj-cortex265.RDS"</span><span class="op">)</span> <span class="co"># new</span></code></pre></div>
<p>In order to work with the functions of <code>monocle3</code> you need to compile it’s core object: the cell_data-set. You can do that with ease with <code><a href="../reference/transformSpataToCDS.html">transformSpataToCDS()</a></code>. It is a wrapper around all of <code>monocle3</code>’s pre processing functions, takes your spata-object of interest and creates a cell_data_set from it’s count matrix. Convenient argument input via named lists and defensive error handling via <code><a href="https://rdrr.io/r/base/conditions.html">base::tryCatch()</a></code> allow more experienced users to tweak the pre processing results. For the sake of simplicity we stick to the default options throughout this example.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># compile a cell-data-set</span>
<span class="va">cortex_cds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/transformSpataToCDS.html">transformSpataToCDS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<p>To keep track of corresponding objects of different platforms you can store the directories under which you save them in the spata-object. To obtain the directories that are currently stored use <code><a href="../reference/getDefaultInstructions.html">getDirectoryInstructions()</a></code>. Add the directory to the corresponding cell_data_set via <code>adjustDirectoryInstruction()</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">spata_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/adjustDefaultInstructions.html">adjustDirectoryInstructions</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>,
                                            to <span class="op">=</span> <span class="st">"cell_data_set"</span>,
                                            directory_new <span class="op">=</span> <span class="st">"data/cds-cortex265.RDS"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># obtain the directories that are currently stored in the spata-object</span>
<span class="fu"><a href="../reference/getDefaultInstructions.html">getDirectoryInstructions</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code>## $cell_data_set
## [1] "data/cds-cortex265.RDS"
## 
## $seurat_object
## [1] "data/seurat-obj-cortex265.RDS"
## 
## $spata_object
## [1] "data/spata-obj-cortex265.RDS"</code></pre>
<p>To save or load a corresponding cell_data_set under the default directory stored in the spata-object use <code><a href="../reference/saveSpataObject.html">saveCorrespondingCDS()</a></code> and <code><a href="../reference/loadSpataObject.html">loadCorrespondingCDS()</a></code></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># use the directory information of the spata-object to conveniently save the cell_data_set</span>
<span class="fu"><a href="../reference/saveSpataObject.html">saveCorrespondingCDS</a></span><span class="op">(</span>cds <span class="op">=</span> <span class="va">cds</span>, object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># use the directory information of the spata-object to conveniently load the cell_data_set</span>
<span class="va">cortex_cds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/loadSpataObject.html">loadCorrespondingCDS</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span> </code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">spata_plot</span> <span class="op">&lt;-</span> 
  <span class="fu"><a href="../reference/plotSurface.html">plotSurface</a></span><span class="op">(</span><span class="va">spata_obj</span>, color_by <span class="op">=</span> <span class="st">"seurat_clusters"</span>, pt_clrp <span class="op">=</span> <span class="st">"npg"</span>, pt_size <span class="op">=</span> <span class="fl">1.4</span><span class="op">)</span> <span class="op">+</span> 
      <span class="fu"><a href="../reference/legendBottom.html">legendBottom</a></span><span class="op">(</span><span class="op">)</span>


<span class="co"># SPATAS transform functions pass the feature data to the compiled object which makes </span>
<span class="co"># all features computed by yourself available, e.g. for monocle3::plot_cells()</span>
<span class="va">monocle_plot</span> <span class="op">&lt;-</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/monocle3/man/plot_cells.html">plot_cells</a></span><span class="op">(</span><span class="va">cortex_cds</span>,
             reduction_method <span class="op">=</span> <span class="st">"UMAP"</span>,
             color_cells_by <span class="op">=</span> <span class="st">"seurat_clusters"</span> <span class="co"># visualize a transferred variable</span>
             <span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="../reference/scale_color_add_on.html">scale_color_add_on</a></span><span class="op">(</span>aes <span class="op">=</span> <span class="st">"color"</span>, variable <span class="op">=</span> <span class="st">"discrete"</span>, clrp <span class="op">=</span> <span class="st">"npg"</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="../reference/legendBottom.html">legendNone</a></span><span class="op">(</span><span class="op">)</span> 

<span class="co"># output 1 &amp; 2 </span>
<span class="va">spata_plot</span>

<span class="va">monocle_plot</span></code></pre></div>
<div class="figure">
<img src="spata-v2-platform-compatibility_files/figure-html/plot-monocle-1.png" alt="Figure 2.1 SPATA's and monocle3' basic plotting results" width="100%"><img src="spata-v2-platform-compatibility_files/figure-html/plot-monocle-2.png" alt="Figure 2.1 SPATA's and monocle3' basic plotting results" width="100%"><p class="caption">
Figure 2.1 SPATA’s and monocle3’ basic plotting results
</p>
</div>
<div id="example-pseudotime" class="section level3">
<h3 class="hasAnchor">
<a href="#example-pseudotime" class="anchor"></a>Example: Pseudotime</h3>
<p>If you haven’t done it already during <code>cortex_cds</code>’s compilation you can use <code>moncle3::order_cells()</code> to work with <a href="https://cole-trapnell-lab.github.io/monocle3/docs/trajectories/">pseudotime</a> - a measurement of how much progress cells have made through a process such as cell differentiation.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># the pseudotime values for all cells/barcode-spots are obtained via</span>
<span class="va">pseudotime_vec</span> <span class="op">&lt;-</span> <span class="fu">monocle3</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/monocle3/man/pseudotime.html">pseudotime</a></span><span class="op">(</span><span class="va">cortex_cds</span><span class="op">)</span>

<span class="co"># subset output</span>
<span class="va">pseudotime_vec</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></code></pre></div>
<pre><code>## AAACAAGTATCTCCCA-1 AAACACCAATAACTGC-1 AAACAGTGTTCCTGGG-1 AAACATTTCCCGGATT-1 
##        17.99675712         0.05947085         0.05946980        13.00497429 
## AAACCCGAACGAAATC-1 AAACCGGAAATGTTAA-1 AAACCGGGTAGGTACC-1 AAACCGTTCGTCCAGG-1 
##        19.24685707        18.82008481         1.21056350         0.05946874 
## AAACCTAAGCAGCCGG-1 AAACCTCATGAAGTTG-1 
##         9.59278155         8.54330799</code></pre>
<p>The output of <code><a href="https://rdrr.io/pkg/monocle3/man/pseudotime.html">monocle3::pseudotime()</a></code> is a named vector of numeric values. The numeric value refers to a barcode-spot’s pseudotime value and the name to it’s barcode. We can easily join this information to our original spata-object with a little bit of data wrangling and the <code><a href="../reference/addFeatures.html">addFeatures()</a></code>-function.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># 1. convert to data.frame </span>
<span class="va">pseudotime_df</span> <span class="op">&lt;-</span> 
  <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">pseudotime_vec</span><span class="op">)</span>

<span class="co"># output</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">pseudotime_df</span><span class="op">)</span></code></pre></div>
<pre><code>##                    pseudotime_vec
## AAACAAGTATCTCCCA-1    17.99675712
## AAACACCAATAACTGC-1     0.05947085
## AAACAGTGTTCCTGGG-1     0.05946980
## AAACATTTCCCGGATT-1    13.00497429
## AAACCCGAACGAAATC-1    19.24685707
## AAACCGGAAATGTTAA-1    18.82008481</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># 2. create a barcodes-key variable and rename the pseudotime variable </span>
<span class="va">feature_df</span> <span class="op">&lt;-</span> 
  <span class="fu">magrittr</span><span class="fu">::</span><span class="fu"><a href="https://magrittr.tidyverse.org/reference/aliases.html">set_colnames</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">pseudotime_df</span>, value <span class="op">=</span> <span class="st">"Pseudotime"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"barcodes"</span><span class="op">)</span>

<span class="co"># output</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">feature_df</span><span class="op">)</span></code></pre></div>
<pre><code>##             barcodes  Pseudotime
## 1 AAACAAGTATCTCCCA-1 17.99675712
## 2 AAACACCAATAACTGC-1  0.05947085
## 3 AAACAGTGTTCCTGGG-1  0.05946980
## 4 AAACATTTCCCGGATT-1 13.00497429
## 5 AAACCCGAACGAAATC-1 19.24685707
## 6 AAACCGGAAATGTTAA-1 18.82008481</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># 3. add to spata-object via </span>
<span class="va">spata_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addFeatures.html">addFeatures</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>,
                         feature_names <span class="op">=</span> <span class="st">"Pseudotime"</span>, 
                         feature_df <span class="op">=</span> <span class="va">feature_df</span>, 
                         overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>The feature <em>Pseudotime</em> is now a valid feature in our spata-object and thus accessible via all functions in <strong>SPATA2</strong> that take numeric features as an input option.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># visualize pseudotime on the surface</span>
<span class="fu"><a href="../reference/plotSurface.html">plotSurface</a></span><span class="op">(</span><span class="va">spata_obj</span>, color_by <span class="op">=</span> <span class="st">"Pseudotime"</span>, smooth <span class="op">=</span> <span class="cn">TRUE</span>, smooth_span <span class="op">=</span> <span class="fl">0.2</span>, pt_size <span class="op">=</span> <span class="fl">1.8</span><span class="op">)</span> </code></pre></div>
<div class="figure">
<img src="spata-v2-platform-compatibility_files/figure-html/demonstrate-ptime-1.png" alt="Figure 3.2 Visualizing monocle3-pseudotime results with SPATA" width="100%"><p class="caption">
Figure 3.2 Visualizing monocle3-pseudotime results with SPATA
</p>
</div>
</div>
</div>
<div id="seurat" class="section level2">
<h2 class="hasAnchor">
<a href="#seurat" class="anchor"></a>3. Seurat</h2>
<p><a href="https://satijalab.org/seurat/vignettes.html"><code>seurat</code></a> provides a variety of computational tools. Initiating a <strong>SPATA2</strong>-object currently relies on most of it’s pre processing functions. If you want to switch from <strong>SPATA2</strong> to <code>Seurat</code> make use of <code><a href="../reference/transformSpataToSeurat.html">transformSpataToSeurat()</a></code>. Similarly to <code><a href="../reference/initiateSpataObject_10X.html">initiateSpataObject_10X()</a></code> and <code><a href="../reference/initiateSpataObject_CountMtr.html">initiateSpataObject_CountMtr()</a></code> it is a wrapper around several preprocessing functions although it passes the feature-data of your spata-object to it in order not to loose your progress. It works in a similar fashion as <code><a href="../reference/transformSpataToCDS.html">transformSpataToCDS()</a></code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cortex_seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/transformSpataToSeurat.html">transformSpataToSeurat</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<p>You can now continue your analysis with Seurat. Note that <strong>SPATA2</strong> handles the image plotting differently which results in mirror-inverted plots.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># continue analysis adn plotting with seurat</span>
<span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialFeaturePlot</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">cortex_seurat</span>, features <span class="op">=</span> <span class="st">"Pseudotime"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-platform-compatibility_files/figure-html/seurat-spatial-plot-1.png" alt="Figure 4.1 Continue plotting and analysis with Seurat" width="100%"><p class="caption">
Figure 4.1 Continue plotting and analysis with Seurat
</p>
</div>
<p>Note that the meta.data slot of the seurat-object contains the barcodes- and the sample-variable of the spata-object it originally derived from.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># output</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">cortex_seurat</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></code></pre></div>
<pre><code>##                       orig.ident nCount_Spatial nFeature_Spatial sample
## AAACAAGTATCTCCCA-1 SeuratProject           1809             1287   C265
## AAACACCAATAACTGC-1 SeuratProject           1772             1235   C265
## AAACAGTGTTCCTGGG-1 SeuratProject           1445             1057   C265
## AAACATTTCCCGGATT-1 SeuratProject           2198             1476   C265
## AAACCCGAACGAAATC-1 SeuratProject           2814             1811   C265
## AAACCGGAAATGTTAA-1 SeuratProject           2900             1880   C265
##                    percent.mt percent.RB Spatial_snn_res.0.8 seurat_clusters
## AAACAAGTATCTCCCA-1   1.492537          0                   2               2
## AAACACCAATAACTGC-1   1.523702          0                   5               5
## AAACAGTGTTCCTGGG-1   1.453287          0                   4               4
## AAACATTTCCCGGATT-1   1.865332          0                   2               2
## AAACCCGAACGAAATC-1   1.279318          0                   0               0
## AAACCGGAAATGTTAA-1   1.586207          0                   0               0
##                    segmentation  Pseudotime mncl_cluster_umap
## AAACAAGTATCTCCCA-1         none 17.99675712                 1
## AAACACCAATAACTGC-1         none  0.05947085                 1
## AAACAGTGTTCCTGGG-1         none  0.05946980                 1
## AAACATTTCCCGGATT-1         none 13.00497429                 1
## AAACCCGAACGAAATC-1         none 19.24685707                 1
## AAACCGGAAATGTTAA-1         none 18.82008481                 1</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Jan Kueckelhaus, Dieter-Henrik Heiland.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
