<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Copy Number Variations (CNV) • SPATA2</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Copy Number Variations (CNV)">
<meta property="og:description" content="SPATA2">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SPATA2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/spata-v2-introduction-and-philosophy.html">About</a>
</li>
<li>
  <a href="../articles/spata-v2-installation.html">Installation</a>
</li>
<li>
  <a href="../articles/spata-v2-terminology.html">Terminology</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Around the Spata-Object</li>
    <li>
      <a href="../articles/spata-v2-object-initiation-and-manipulation.html">Object Set Up and Manipulation</a>
    </li>
    <li>
      <a href="../articles/spata-v2-extract-add-and-join-data.html">Extracting, Adding and Joining Data</a>
    </li>
    <li>
      <a href="../articles/spata-v2-platform-compatibility.html">Integrating Different Platforms</a>
    </li>
    <li class="dropdown-header">Analysis &amp; Computation</li>
    <li>
      <a href="../articles/spata-v2-autoencoder.html">Autoencoder Denoising</a>
    </li>
    <li>
      <a href="../articles/spata-v2-cnv-analysis.html">Copy Number Variations</a>
    </li>
    <li>
      <a href="../articles/spata-v2-segmentation-and-clustering.html">Spatial Segmentation &amp; Clustering</a>
    </li>
    <li>
      <a href="../articles/spata-v2-de-analysis.html">Differential Expression Analysis</a>
    </li>
    <li>
      <a href="../articles/sapta-v2-gsea.html">Gene Set Enrichment Analysis</a>
    </li>
    <li>
      <a href="../articles/spata-v2-trajectory-analysis.html">Spatial Trajectories</a>
    </li>
    <li class="dropdown-header">Visualization</li>
    <li>
      <a href="../articles/spata-v2-plotting-surface.html">Surface Plotting</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-dimensional-reduction.html">Dimensional Reduction</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-descriptive-statistics.html">Descriptive Statistics</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-miscellaneous.html">Miscellaneous</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="spata-v2-cnv-analysis_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Copy Number Variations (CNV)</h1>
            
      
      
      <div class="hidden name"><code>spata-v2-cnv-analysis.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">1. Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>InferCNV is used to identify large-scale chromosomal copy number alterations in single cell RNA-Seq data including gains or deletions of chromosomes or large segments of such. Several publications have leveraged this technique. (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4123637/" class="external-link">Patel et al., 2014</a>, <a href="https://pubmed.ncbi.nlm.nih.gov/27124452/" class="external-link">Tirosh et al., 2016</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5519096/" class="external-link">Venteicher et al., 2017</a> )</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://themilolab.com/" class="external-link">SPATA2</a></span><span class="op">)</span>

<span class="co"># load glioblastoma sample</span>
<span class="va">spata_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/loadSpataObject.html">loadSpataObject</a></span><span class="op">(</span><span class="st">"data/spata-obj-gbm275.RDS"</span><span class="op">)</span></code></pre></div>
<p><strong>SPATA2</strong> implements the package <a href="https://github.com/broadinstitute/inferCNV/wiki" class="external-link"><code>infercnv</code></a> published and maintained by Broadinstitute and allows to integrate this technique in your workflow of analysing spatial trancsriptomic data derived from malignancies.</p>
</div>
<div class="section level2">
<h2 id="running-cnv-analysis">2. Running CNV-Analysis<a class="anchor" aria-label="anchor" href="#running-cnv-analysis"></a>
</h2>
<p>As with all other functions prefixed with <code>run*()</code> the function <code><a href="../reference/runCnvAnalysis.html">runCnvAnalysis()</a></code> is a wrapper around all necessary functions needed to conduct copy-number-variation-analysis. The results needed for subsequent analysis steps are stored in the specified spata-object (slot: <span class="citation">@cnv</span>). The infercnv-object is stored in the folder specified in the argument <code>directory_cnv_folder</code>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r code-chunk"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>spata_obj &lt;-</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="st">  </span><span class="kw">runCnvAnalysis</span>(</span>
<span id="cb2-3"><a href="#cb2-3"></a>    <span class="dt">object =</span> spata_obj</span>
<span id="cb2-4"><a href="#cb2-4"></a>    <span class="dt">directory_cnv_folder =</span> <span class="st">"data-gmb275/cnv-results"</span> <span class="co"># example directory</span></span>
<span id="cb2-5"><a href="#cb2-5"></a>    <span class="dt">cnv_prefix =</span> <span class="st">"Chr"</span></span>
<span id="cb2-6"><a href="#cb2-6"></a>    )</span></code></pre></div>
<p>If your desired set up deviates from the default you can reach any function of the inferncnv-pipeline by entering it’s name as an argument and specify it’s input as a list of arguments with which you want it to be called.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r code-chunk"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># change input</span></span>
<span id="cb3-2"><a href="#cb3-2"></a>spata_obj &lt;-<span class="st"> </span></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="st">  </span><span class="kw">runCnvAnalysis</span>(</span>
<span id="cb3-4"><a href="#cb3-4"></a>    <span class="dt">object =</span> spata_obj</span>
<span id="cb3-5"><a href="#cb3-5"></a>    <span class="dt">directory_cnv_folder =</span> <span class="st">"data-gmb275/cnv-results"</span> <span class="co"># example directory to </span></span>
<span id="cb3-6"><a href="#cb3-6"></a>    <span class="dt">clear_noise_via_ref_mean_sd =</span> <span class="kw">list</span>(<span class="dt">sd_amplifier =</span> <span class="dv">2</span>)</span>
<span id="cb3-7"><a href="#cb3-7"></a>    )</span></code></pre></div>
<p>Copy number variation analysis requires reference data. This includes a count matrix from healthy tissue, an annotation file as well as a data.frame that contains information about the chromosome positions. We provide reference data in the list <code><a href="../reference/cnv_ref.html">SPATA2::cnv_ref</a></code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cnv_ref</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "annotation" "mtr"        "regions"</span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">cnv_ref</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##            Length   Class      Mode   </span>
<span class="co">## annotation        1 data.frame list   </span>
<span class="co">## mtr        71067022 -none-     numeric</span>
<span class="co">## regions           4 data.frame list</span></code></pre>
<p><code><a href="../reference/runCnvAnalysis.html">runCnvAnalysis()</a></code> defaults to the content of this list. The documentation of <code><a href="../reference/runCnvAnalysis.html">runCnvAnalysis()</a></code> contains a detailed description of the requirements each reference input must meet in order for the function to work.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># example on how to use the functions arguments if you want to specify the reference (ref_*)</span>
<span class="fu"><a href="../reference/runCnvAnalysis.html">runCnvAnalysis</a></span><span class="op">(</span>
  object <span class="op">=</span> <span class="va">spata_obj</span>,
  directory_cnv_folder <span class="op">=</span> <span class="st">"data-gmb275/cnv-results"</span>, <span class="co"># example directory to </span>
  ref_annotation <span class="op">=</span> <span class="va">cnv_ref</span><span class="op">$</span><span class="va">annotation</span>, 
  ref_mtr <span class="op">=</span> <span class="va">cnv_ref</span><span class="op">$</span><span class="va">mtr</span>, 
  ref_regions <span class="op">=</span> <span class="va">cnv_ref</span><span class="op">$</span><span class="va">regions</span>
<span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="cnv-results">3. CNV-Results<a class="anchor" aria-label="anchor" href="#cnv-results"></a>
</h2>
<p>The results are stored in a list inside the spata-object. This list can be obtained via <code><a href="../reference/getCnvResults.html">getCnvResults()</a></code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cnv_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getCnvResults.html">getCnvResults</a></span><span class="op">(</span><span class="va">spata_obj</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cnv_results</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "prefix"      "gene_pos_df" "cnv_df"      "cnv_mtr"     "regions_df" </span>
<span class="co">## [6] "dim_red"     "clustering"</span></code></pre>
</div>
<div class="section level2">
<h2 id="visualization">4. Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<p>The gains and losses of chromosomal segments can be displayed via <code><a href="../reference/plotCnvResults.html">plotCnvResults()</a></code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotCnvResults.html">plotCnvResults</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-cnv-analysis_files/figure-html/plotCnvResults()-1.png" alt="Figure 4.1 Copy Number Alterations for the whole sample." width="100%"><p class="caption">
Figure 4.1 Copy Number Alterations for the whole sample.
</p>
</div>
<p>If you want to visualize the results across certain groups to identify subclones make use of the <code>across</code>-argument.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotSurface.html">plotSurface</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, color_by <span class="op">=</span> <span class="st">"seurat_clusters"</span><span class="op">)</span>

<span class="fu"><a href="../reference/plotCnvResults.html">plotCnvResults</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, across <span class="op">=</span> <span class="st">"seurat_clusters"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-cnv-analysis_files/figure-html/plotCnvResults()-across-1.png" alt="Figure 4.2 Copy Number Alterations across specific groups." width="50%"><img src="spata-v2-cnv-analysis_files/figure-html/plotCnvResults()-across-2.png" alt="Figure 4.2 Copy Number Alterations across specific groups." width="50%"><p class="caption">
Figure 4.2 Copy Number Alterations across specific groups.
</p>
</div>
<p>Apart from that you obtain a heatmap that is directly plotted into the folder specified with argument <code>directory_cnv_results</code>.</p>
<div class="figure">
<img src="images/cnv-heatmap.png" alt="Figure 4.3 CNV-Heatmap" width="100%"><p class="caption">
Figure 4.3 CNV-Heatmap
</p>
</div>
<p>Note: We are currently encountering problems with the function <code>infercnv:::plot_cnv()</code>. It does not prevent the copy number variation analysis. Sometimes, however, the heatmap is not plotted. Checkout your R console for hints in how far the function did not work properly. We are working on a solution for that. If you encounter warnings raised from <code><a href="../reference/runCnvAnalysis.html">runCnvAnalysis()</a></code> <a href="https://github.com/theMILOlab/SPATA2/issues" class="external-link">let us know</a>.</p>
<p>The numeric values by which the copy number variations within each barcode-spot are represented are immediately transferred to the spata-object’s feature data and are thus accessible for all functions that work with numeric variables. The character string specified in argument <code>cnv_prefix</code> combined with the chromosomes number determines the name by which you can refer to these feature variables.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># cnv feature names</span>
<span class="fu"><a href="../reference/getCnvFeatureNames.html">getCnvFeatureNames</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  [1] "Chr1"  "Chr1"  "Chr2"  "Chr2"  "Chr3"  "Chr3"  "Chr4"  "Chr4"  "Chr5" </span>
<span class="co">## [10] "Chr5"  "Chr6"  "Chr6"  "Chr7"  "Chr7"  "Chr8"  "Chr8"  "Chr9"  "Chr9" </span>
<span class="co">## [19] "Chr10" "Chr10" "Chr11" "Chr11" "Chr12" "Chr12" "Chr13" "Chr13" "Chr14"</span>
<span class="co">## [28] "Chr14" "Chr15" "Chr15" "Chr16" "Chr16" "Chr17" "Chr17" "Chr18" "Chr18"</span>
<span class="co">## [37] "Chr19" "Chr19" "Chr20" "Chr20" "Chr21" "Chr21" "Chr22" "Chr22"</span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># are part of all feature names</span>
<span class="fu"><a href="../reference/getFeatureNames.html">getFeatureNames</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##            numeric            integer            numeric            numeric </span>
<span class="co">##   "nCount_Spatial" "nFeature_Spatial"       "percent.mt"       "percent.RB" </span>
<span class="co">##             factor          character             factor             factor </span>
<span class="co">##  "seurat_clusters"     "segmentation"      "Leiden_UMAP"          "cluster" </span>
<span class="co">##            logical            numeric            numeric            numeric </span>
<span class="co">##             "keep"             "Chr0"             "Chr1"            "Chr10" </span>
<span class="co">##            numeric            numeric            numeric            numeric </span>
<span class="co">##            "Chr11"            "Chr12"            "Chr13"            "Chr14" </span>
<span class="co">##            numeric            numeric            numeric            numeric </span>
<span class="co">##            "Chr15"            "Chr16"            "Chr17"            "Chr18" </span>
<span class="co">##            numeric            numeric            numeric            numeric </span>
<span class="co">##            "Chr19"             "Chr2"            "Chr20"            "Chr21" </span>
<span class="co">##            numeric            numeric            numeric            numeric </span>
<span class="co">##            "Chr22"            "Chr23"             "Chr3"             "Chr4" </span>
<span class="co">##            numeric            numeric            numeric            numeric </span>
<span class="co">##             "Chr5"             "Chr6"             "Chr7"             "Chr8" </span>
<span class="co">##            numeric </span>
<span class="co">##             "Chr9"</span></code></pre>
<p>Figure 4.3 as well as figure 4.1 indicate that the glioblastoma sample contains losses in chromosome 10 and gains in chromosome 7 and that these alterations affect cluster <em>1</em> particularly strong while disregarding cluster <em>2</em> (upper-left) of the grouping variable <em>seurat_clusters</em>. Use <code><a href="../reference/plotSurface.html">plotSurface()</a></code> to visualize these subclones on your sample’s surface.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotSurface.html">plotSurface</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, color_by <span class="op">=</span> <span class="st">"Chr7"</span>, pt_clrsp <span class="op">=</span> <span class="st">"Reds 3"</span>, c1 <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>

<span class="fu"><a href="../reference/plotSurface.html">plotSurface</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, color_by <span class="op">=</span> <span class="st">"Chr10"</span>, pt_clrsp <span class="op">=</span> <span class="st">"Blues 3"</span>, rev <span class="op">=</span> <span class="cn">FALSE</span>, c1 <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-cnv-analysis_files/figure-html/plot-cnv-on-surface-1.png" alt="Figure 4.4 Spatial visualization of areas that are affected by chromosomal alterations." width="50%"><img src="spata-v2-cnv-analysis_files/figure-html/plot-cnv-on-surface-2.png" alt="Figure 4.4 Spatial visualization of areas that are affected by chromosomal alterations." width="50%"><p class="caption">
Figure 4.4 Spatial visualization of areas that are affected by chromosomal alterations.
</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jan Kueckelhaus, Dieter-Henrik Heiland.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
