<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Differential Expression Analysis • SPATA2</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Differential Expression Analysis">
<meta property="og:description" content="SPATA2">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SPATA2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/spata-v2-introduction-and-philosophy.html">About</a>
</li>
<li>
  <a href="../articles/spata-v2-installation.html">Installation</a>
</li>
<li>
  <a href="../articles/spata-v2-terminology.html">Terminology</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Around the Spata-Object</li>
    <li>
      <a href="../articles/spata-v2-object-initiation-and-manipulation.html">Object Set Up and Manipulation</a>
    </li>
    <li>
      <a href="../articles/spata-v2-extract-add-and-join-data.html">Extracting, Adding and Joining Data</a>
    </li>
    <li>
      <a href="../articles/spata-v2-platform-compatibility.html">Integrating Different Platforms</a>
    </li>
    <li class="dropdown-header">Analysis &amp; Computation</li>
    <li>
      <a href="../articles/spata-v2-autoencoder.html">Autoencoder Denoising</a>
    </li>
    <li>
      <a href="../articles/spata-v2-de-analysis.html">Differential Expression Analysis</a>
    </li>
    <li>
      <a href="../articles/spata-v2-segmentation-and-clustering.html">Spatial Segmentation &amp; Clustering</a>
    </li>
    <li>
      <a href="../articles/spata-v2-trajectory-analysis.html">Spatial Trajectories</a>
    </li>
    <li class="dropdown-header">Visualization</li>
    <li>
      <a href="../articles/spata-v2-plotting-surface.html">Surface Plotting</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-dimensional-reduction.html">Dimensional Reduction</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-descriptive-statistics.html">Descriptive Statistics</a>
    </li>
    <li>
      <a href="../articles/spata-v2-plotting-miscellaneous.html">Miscellaneous</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="spata-v2-de-analysis_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Differential Expression Analysis</h1>
            
      
      
      <div class="hidden name"><code>spata-v2-de-analysis.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>1. Introduction</h2>
<p>Differential expression analysis aims to discover quantitative changes in gene-expression levels between defined experimental groups. In <strong>SPATA2</strong> these experimental groups are defined inside the feature data. More precisely: Every discrete variable that is part of the spata-object’s feature data assigns every sample’s barcode-spot to such an experimental group. This includes all spata-intern generated groups such as <em>segmentation</em> and <em>clustering</em> as well as any other discrete feature of your own extracted analysis that has been added via <code><a href="../reference/addFeatures.html">addFeatures()</a></code>. See <a href="spata-v2-extern-analysis.html">Extract, Add &amp; Join data</a> to learn how to do that with ease.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load packages </span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://themilolab.com/">SPATA2</a></span><span class="op">)</span> 
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>

<span class="va">spata_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/loadSpataObject.html">loadSpataObject</a></span><span class="op">(</span><span class="st">"data/spata-obj-gbm275.RDS"</span><span class="op">)</span></code></pre></div>
<p>As mentioned in previous chapters <strong>SPATA2</strong> makes use of some well written functions of the package <a href="https://satijalab.org/seurat/"><code>Seurat</code></a> which currently define the <em>gold-standard</em> of statistical scRNA-seq analysis in R. The function <code><a href="../reference/runDeAnalysis.html">runDeAnalysis()</a></code>is a wrapper around <code><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html">Seurat::FindAllMarkers()</a></code> which finds the marker genes of the identity groups you denote with the <code>across</code>-argument. In the tutorial on <a href="spata-v2-segmentation-and-clustering.html">segmentation and clustering</a> in <em>SPATA</em> we created several grouping variables. Throughout this example we will try and find the differently expressed genes across the clusters of the <em>"“Leiden_UMAP”"</em> variable.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotSurface.html">plotSurface</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
            color_by <span class="op">=</span> <span class="st">"Leiden_UMAP"</span>, 
            pt_clrp <span class="op">=</span> <span class="st">"jama"</span>, 
            pt_size <span class="op">=</span> <span class="fl">1.9</span>
            <span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-de-analysis_files/figure-html/example-visualization-1.png" alt="Figure 1.1 The chosen algorithm grouped the barcode spots in seven different clusters for which we are interested in their marker genes." width="100%"><p class="caption">
Figure 1.1 The chosen algorithm grouped the barcode spots in seven different clusters for which we are interested in their marker genes.
</p>
</div>
</div>
<div id="running-the-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#running-the-analysis" class="anchor"></a>2. Running the analysis</h2>
<p>The output of <code><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html">Seurat::FindAllMarkers()</a></code> is a data.frame in which each row corresponds to a gene that turned out to be a marker gene for one of the identity groups. Additional variables provide information about it’s p-value, adjusted p-value, logFold change etc. <code><a href="../reference/runDeAnalysis.html">runDeAnalysis()</a></code> does not alter that output but stores it in the spata-object, more precisely in the <code>@dea</code> slot, short form for differential-expression-analysis. This feature allows to conveniently iterate over several methods like <em>‘wilcox’</em> or <em>‘bimod’</em> without loosing track of the results nor of the object- or variable-belonging. With <code><a href="../reference/printDeaOverview.html">printDeaOverview()</a></code> you can check across which groups and according to which method de-analysis has already been conducted.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">spata_obj</span> <span class="op">&lt;-</span> 
  <span class="fu"><a href="../reference/runDeAnalysis.html">runDeAnalysis</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>,
                across <span class="op">=</span> <span class="st">"Leiden_UMAP"</span>, <span class="co"># across which identity groups</span>
                method_de <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"wilcox"</span>, <span class="st">"bimod"</span><span class="op">)</span> <span class="co"># with which methods</span>
                <span class="op">)</span>

<span class="co"># get an overview about the de-analysis results stored in your object</span>
<span class="fu"><a href="../reference/printDeaOverview.html">printDeaOverview</a></span><span class="op">(</span><span class="va">spata_obj</span><span class="op">)</span></code></pre></div>
<pre><code>## De-analysis has been performed on grouping variables:
## 
## - 'segmentation' with methods: wilcox
## - 'seurat_clusters' with methods: wilcox
## - 'Leiden_UMAP' with methods: wilcox' and 'bimod</code></pre>
</div>
<div id="extracting-results" class="section level2">
<h2 class="hasAnchor">
<a href="#extracting-results" class="anchor"></a>3. Extracting results</h2>
<p>There are two main functions with which you can manually extract the de-analysis results desired. First, <code><a href="../reference/getDeaResultsDf.html">getDeaResultsDf()</a></code> returns the original resulting data.frame of <code><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html">Seurat::FindAllMarkers()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getDeaResultsDf.html">getDeaResultsDf</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
                across <span class="op">=</span> <span class="st">"Leiden_UMAP"</span>, 
                method_de <span class="op">=</span> <span class="st">"wilcox"</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 5,971 x 7
##        p_val avg_logFC pct.1 pct.2 p_val_adj Leiden_UMAP gene  
##        &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;       &lt;chr&gt; 
##  1 1.66e-  4     299.  0.151 0.105 1.00e+  0 Cluster 1   NPY   
##  2 3.26e- 57     264.  0.3   0.093 7.03e- 53 Cluster 1   TAC1  
##  3 7.68e- 34      92.6 0.173 0.05  1.66e- 29 Cluster 1   SST   
##  4 2.84e- 17      79.6 1     0.999 6.12e- 13 Cluster 1   GFAP  
##  5 1.38e- 12      36.3 1     0.999 2.97e-  8 Cluster 1   S100A6
##  6 1.86e-195      28.4 0.964 0.761 4.02e-191 Cluster 1   UCHL1 
##  7 0.             17.1 0.966 0.461 0.        Cluster 1   SNAP25
##  8 1.99e- 49      16.5 0.927 0.845 4.28e- 45 Cluster 1   TUBB2A
##  9 5.72e- 14      15.7 0.993 0.982 1.23e-  9 Cluster 1   TUBA1B
## 10 3.98e- 23      15.6 0.988 0.967 8.57e- 19 Cluster 1   UBB   
## # ... with 5,961 more rows</code></pre>
<p>The de-data.frame can be additionally processed such that the following steps are performed for every identity group. <code><a href="../reference/getDeaResultsDf.html">getDeaResultsDf()</a></code> does that by calling the function <code><a href="../reference/filterDeaDf.html">filterDeaDf()</a></code> which</p>
<ol style="list-style-type: decimal">
<li><p>discards genes with avg_logFC-values that are either infinite or negative</p></li>
<li><p>discards genes with adjusted p-values above the threshold set with <code>max_adj_pval</code></p></li>
<li><p>slices the data.frame in order that for every unique cluster of the cluster-variable:</p></li>
</ol>
<ul>
<li><p>the n genes with the highest avg_logFC-values are kept where n = <code>n_highest_lfc</code></p></li>
<li><p>the n genes with the lowest p_val_adj-values are kept where n = <code>n_lowest_pval</code></p></li>
</ul>
<ol start="4" style="list-style-type: decimal">
<li>arranges the genes according to the highest avg_logFC-values</li>
</ol>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getDeaResultsDf.html">getDeaResultsDf</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
                across <span class="op">=</span> <span class="st">"Leiden_UMAP"</span>, 
                method_de <span class="op">=</span> <span class="st">"wilcox"</span>, 
                max_adj_pval <span class="op">=</span> <span class="fl">0.025</span>, <span class="co"># of every cluser take genes with an adj. p-value of 0.025 or lower</span>
                n_lowest_pval <span class="op">=</span> <span class="fl">20</span><span class="op">)</span> <span class="co"># from the remaining genes take only to 'top 20')</span></code></pre></div>
<pre><code>## # A tibble: 105 x 7
##        p_val avg_logFC pct.1 pct.2 p_val_adj Leiden_UMAP gene   
##        &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;       &lt;chr&gt;  
##  1 1.86e-195     28.4  0.964 0.761 4.02e-191 Cluster 1   UCHL1  
##  2 0.            17.1  0.966 0.461 0.        Cluster 1   SNAP25 
##  3 1.35e-263     15.0  0.805 0.229 2.91e-259 Cluster 1   SYT1   
##  4 1.88e-165     13.6  0.9   0.619 4.05e-161 Cluster 1   ENC1   
##  5 4.46e-267     12.2  0.809 0.232 9.61e-263 Cluster 1   VSNL1  
##  6 3.03e-187     11.7  0.877 0.458 6.54e-183 Cluster 1   OLFM1  
##  7 5.72e-182     11.6  0.552 0.112 1.23e-177 Cluster 1   HPCA   
##  8 2.35e-139     11.4  0.895 0.608 5.07e-135 Cluster 1   YWHAH  
##  9 6.61e-191     10.0  0.911 0.511 1.42e-186 Cluster 1   RTN1   
## 10 5.66e-219      9.74 0.715 0.196 1.22e-214 Cluster 1   SLC17A7
## # ... with 95 more rows</code></pre>
</div>
<div id="plotting-results" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-results" class="anchor"></a>4. Plotting results</h2>
<div id="heatmap" class="section level3">
<h3 class="hasAnchor">
<a href="#heatmap" class="anchor"></a>4.1 Heatmap</h3>
<p>The gold standard of gene expression visualization is the classical heatmap. <code><a href="../reference/plotDeaHeatmap.html">plotDeaHeatmap()</a></code> does just that: visualizing the results of your de-analysis by refering to the results you would extract with <code><a href="../reference/getDeaResultsDf.html">getDeaResultsDf()</a></code>. If you don’t want to refer to de-analysis results and just wish to plot certain genes of interest you can do that by passing them to the <code>genes</code>-argument.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cluster_of_interest</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Cluster 2"</span>, <span class="st">"Cluster 3"</span>, <span class="st">"Cluster 6"</span><span class="op">)</span>

<span class="fu"><a href="../reference/plotDeaHeatmap.html">plotDeaHeatmap</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
               across <span class="op">=</span> <span class="st">"Leiden_UMAP"</span>, <span class="co"># the grouping variable</span>
               across_subset <span class="op">=</span> <span class="va">cluster_of_interest</span>, <span class="co"># the identity groups of interest </span>
               method_de <span class="op">=</span> <span class="st">"wilcox"</span>, <span class="co"># the method with which the results were computed</span>
               max_adj_pval <span class="op">=</span> <span class="fl">0.025</span>, <span class="co"># the adjusted p-value threshold</span>
               n_lowest_pval <span class="op">=</span> <span class="fl">20</span>, 
               n_highest_lfc <span class="op">=</span> <span class="fl">20</span>, 
               clrp <span class="op">=</span> <span class="st">"jama"</span>, 
               fontsize <span class="op">=</span> <span class="fl">8</span><span class="op">)</span> </code></pre></div>
<div class="figure">
<img src="spata-v2-de-analysis_files/figure-html/plotDeaHeatmap()-1.png" alt="Figure 4.1 Heatmap visualizing the differentially expressed genes" width="100%"><p class="caption">
Figure 4.1 Heatmap visualizing the differentially expressed genes
</p>
</div>
</div>
<div id="statistic-related-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#statistic-related-plots" class="anchor"></a>4.2 Statistic related plots</h3>
<p>There are additional ways to visualize the results of your de-analysis. As with almost all plotting functions in <strong>SPATA2</strong> a vector of gene names is necessary for the function to know which genes to plot. <code><a href="../reference/getDeaResultsDf.html">getDeaGenes()</a></code> is the second function to extract de-analysis results and a wrapper around <code><a href="../reference/getDeaResultsDf.html">getDeaResultsDf()</a></code> that returns a vector gene names.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genes_of_interest</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/getDeaResultsDf.html">getDeaGenes</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>,
              across <span class="op">=</span> <span class="st">"Leiden_UMAP"</span>, <span class="co"># the grouping variable</span>
              across_subset <span class="op">=</span> <span class="va">cluster_of_interest</span>, <span class="co"># the identity groups of interest </span>
              method_de <span class="op">=</span> <span class="st">"wilcox"</span>, <span class="co"># the method with which the results were computed</span>
              max_adj_pval <span class="op">=</span> <span class="fl">0.025</span>, <span class="co"># the adjusted p-value threshold</span>
              n_lowest_pval <span class="op">=</span> <span class="fl">10</span>, 
              n_highest_lfc <span class="op">=</span> <span class="fl">10</span>
              <span class="op">)</span>


<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">genes_of_interest</span><span class="op">)</span> <span class="co"># first six</span></code></pre></div>
<pre><code>## Cluster 2 Cluster 2 Cluster 2 Cluster 2 Cluster 2 Cluster 2 
##   "POSTN"  "TMSB4X"   "FABP7"   "RAMP1"  "PTPRZ1"   "S100B"</code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span><span class="op">(</span><span class="va">genes_of_interest</span><span class="op">)</span> <span class="co"># last six</span></code></pre></div>
<pre><code>## Cluster 6 Cluster 6 Cluster 6 Cluster 6 Cluster 6 Cluster 6 
##  "EEF1A1"     "ADM"  "CHI3L1"     "CLU"  "TMSB10"     "UBC"</code></pre>
<p>A vector of gene names as returned by <code><a href="../reference/getDeaResultsDf.html">getDeaGenes()</a></code> (or <code><a href="../reference/getGenes.html">getGenes()</a></code> for that matter) is a perfectly valid input for other plotting functions.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">high_threshold_markers</span> <span class="op">&lt;-</span> 
    <span class="fu"><a href="../reference/getDeaResultsDf.html">getDeaGenes</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>,
                across <span class="op">=</span> <span class="st">"Leiden_UMAP"</span>, 
                across_subset <span class="op">=</span> <span class="va">cluster_of_interest</span>, 
                method <span class="op">=</span> <span class="st">"wilcox"</span>, 
                n_lowest_pval <span class="op">=</span> <span class="fl">3</span>
                <span class="op">)</span> <span class="co"># return only the two genes with the lowest p-value for every identity group</span>

<span class="co"># print genes</span>
<span class="va">high_threshold_markers</span></code></pre></div>
<pre><code>## Cluster 2 Cluster 2 Cluster 2 Cluster 3 Cluster 3 Cluster 3 Cluster 6 Cluster 6 
##   "POSTN"   "RAMP1"  "LANCL2"     "B2M"  "HSPA1A"  "MARCH6"     "ADM"   "VEGFA" 
## Cluster 6 
##  "HILPDA"</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># visualize statistics</span>
<span class="fu"><a href="../reference/plotBoxplot.html">plotBoxplot</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
            variables <span class="op">=</span> <span class="va">high_threshold_markers</span>, 
            across <span class="op">=</span> <span class="st">"Leiden_UMAP"</span>, 
            across_subset <span class="op">=</span> <span class="va">cluster_of_interest</span>, 
            clrp <span class="op">=</span> <span class="st">"jama"</span>, 
            nrow <span class="op">=</span> <span class="fl">3</span>
            <span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="../reference/legendBottom.html">legendTop</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-de-analysis_files/figure-html/plotBoxplot()-1.png" alt="Figure 3.2 Plotting descriptive statistics with the analysis results" width="100%"><p class="caption">
Figure 3.2 Plotting descriptive statistics with the analysis results
</p>
</div>
</div>
<div id="surface-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#surface-plots" class="anchor"></a>4.3 Surface plots</h3>
<p>Another way to visualize your results would be to deploy <code><a href="../reference/plotSurfaceComparison.html">plotSurfaceComparison()</a></code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotSurfaceComparison.html">plotSurfaceComparison</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
                      color_by <span class="op">=</span> <span class="va">high_threshold_markers</span>, 
                      smooth <span class="op">=</span> <span class="cn">TRUE</span>, 
                      pt_size <span class="op">=</span> <span class="fl">1</span>
                      <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="../reference/legendBottom.html">legendNone</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># compare to cluster localisation</span>
<span class="fu"><a href="../reference/plotSurface.html">plotSurface</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, 
            color_by <span class="op">=</span> <span class="st">"Leiden_UMAP"</span>, 
            pt_clrp <span class="op">=</span> <span class="st">"jama"</span>
            <span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-de-analysis_files/figure-html/de-surface-1.png" alt="Figure 3.3 Surfaceplots visualizing the differentially expressed genes" width="50%"><img src="spata-v2-de-analysis_files/figure-html/de-surface-2.png" alt="Figure 3.3 Surfaceplots visualizing the differentially expressed genes" width="50%"><p class="caption">
Figure 3.3 Surfaceplots visualizing the differentially expressed genes
</p>
</div>
</div>
<div id="basic-overview" class="section level3">
<h3 class="hasAnchor">
<a href="#basic-overview" class="anchor"></a>4.4 Basic overview</h3>
<p>To obtain a basic overview of the qualities and quantities of your analysis results use <code><a href="../reference/plotDeaGeneCount.html">plotDeaGeneCount()</a></code>, <code><a href="../reference/plotDeaPvalues.html">plotDeaLogFC()</a></code> and <code><a href="../reference/plotDeaPvalues.html">plotDeaPvalues()</a></code>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotDeaGeneCount.html">plotDeaGeneCount</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, across <span class="op">=</span> <span class="st">"Leiden_UMAP"</span>, method_de <span class="op">=</span> <span class="st">"wilcox"</span>, clrp <span class="op">=</span> <span class="st">"jama"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-de-analysis_files/figure-html/plotDeaSummary()-1.png" alt="Figure 3.4 Basic summary of your analysis results" width="100%"><p class="caption">
Figure 3.4 Basic summary of your analysis results
</p>
</div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotDeaPvalues.html">plotDeaLogFC</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>, across <span class="op">=</span> <span class="st">"Leiden_UMAP"</span>, method_de <span class="op">=</span> <span class="st">"wilcox"</span>, clrp <span class="op">=</span> <span class="st">"jama"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-de-analysis_files/figure-html/plotDeaSummary()-2.png" alt="Figure 3.4 Basic summary of your analysis results" width="100%"><p class="caption">
Figure 3.4 Basic summary of your analysis results
</p>
</div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotDeaPvalues.html">plotDeaPvalues</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">spata_obj</span>,
               across <span class="op">=</span> <span class="st">"Leiden_UMAP"</span>,
               method_de <span class="op">=</span> <span class="st">"wilcox"</span>,
               clrp <span class="op">=</span> <span class="st">"jama"</span>,
               plot_type <span class="op">=</span> <span class="st">"density"</span>, 
               scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="spata-v2-de-analysis_files/figure-html/plotDeaSummary()-3.png" alt="Figure 3.4 Basic summary of your analysis results" width="100%"><p class="caption">
Figure 3.4 Basic summary of your analysis results
</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Jan Kueckelhaus, Dieter-Henrik Heiland.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
